---
title: "Expected Counts Data Analyis - edgeR"
author: "Heike Sprenger"
date: "Tuesday, May 05, 2015"
output: html_document
---

## Input: Expected counts per gene (from RSEM)

### Set working directory  
```{r set working directory}
#getwd()
#setwd("D:/work/repos/trost_transcriptomics")
```

[solution for issue with working directory and knitr](https://github.com/yihui/knitr/issues/277)

### Load workspace, packages and scripts
```{r load workspace, message=FALSE}
# load packages
library(knitr)
library(pcaMethods)
library(edgeR)
library(VennDiagram)
library(gplots)
library(RColorBrewer)
library(plyr)
library(reshape2)
library(pander)

# set options for pander
panderOptions('table.split.table', 200)

# set options for knitr
opts_chunk$set(fig.width=5, fig.height=5, cache=FALSE, highlight = TRUE, fig.show="asis")
opts_knit$set(root.dir = 'D:/work/repos/trost_transcriptomics/')

# load workspace
#load("exp_counts_edgeR.RData")
```


### Samplelist processing
```{r child = 'Modify_RNASeq_Samplelist.Rmd'}
# knit('scripts/Modify_RNASeq_Samplelist.Rmd')
```


### Source R functions
```{r source R functions}
source("../libpurzel/func_venn_diagram.R")
source("../libpurzel/func_edgeR_dge.R")
source("../libpurzel/func_pageman_res.R")
source("../libpurzel/func_merge_sig_res_mapman.R")
```


### Load filtered expected counts
```{r load filtered expected conunts}
counts_keep <- read.table("output/counts_keep.txt", sep="\t", header=T)

# rownames of filtered expected counts (DMG identifier)
#rownames_intersect_greenhouse_field <- rownames(counts_keep)
#length(rownames_intersect_greenhouse_field)

# greenhouse
counts_keep_greenhouse <- counts_keep[,which(samplelist_ordered$cultivation=="greenhouse")]

# field
counts_keep_field <- counts_keep[,which(samplelist_ordered$cultivation=="field")]
```


### Load annotation files
```{r load annotation files}
assoc_pgsc <- read.table("data/PGSC_DM_v3.4_g2t2c2p2func_edit.txt")
colnames(assoc_pgsc) <- c("pgsc_dmg", "pgsc_dmt", "pgsc_dmc", "pgsc_dmp", "func")
pander(head(assoc_pgsc))

assoc_pgsc_dmg <- unique(assoc_pgsc[,c(1,5)])

assoc_mapman <- read.table("data/mapman_cleaned_output_new.txt", header = T)
dim(assoc_mapman)
colnames(assoc_mapman)

# merge PGSC annotation table with MapMan table
merge_mapman_pgsc <- merge(assoc_pgsc, assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
colnames(merge_mapman_pgsc)
dim(merge_mapman_pgsc)
write.table(merge_mapman_pgsc, "data/merge_mapman_pgsc.txt", sep="\t")


head(table(merge_mapman_pgsc$BINCODE))

# unique MapMan classifications per gene
merge_mapman_pgsc_dmg <- unique(merge_mapman_pgsc[,c(2,5:7)])
pander(head(merge_mapman_pgsc_dmg))
```


### Melt MapMan table
```{r melt mapman table}
# melt
melt_mapman_pgsc <- melt(merge_mapman_pgsc, id=c("BINCODE")) 
pander(head(melt_mapman_pgsc))

# cast
cast_mapman_pgsc <- dcast(merge_mapman_pgsc, merge_mapman_pgsc$NAME ~ merge_mapman_pgsc$pgsc_dmp) 
dim(cast_mapman_pgsc)

# generate count table
merge_mapman_pgsc_table <- as.data.frame(table(merge_mapman_pgsc$BINCODE)) # generate count table per bincode
merge_mapman_pgsc_table2 <- as.data.frame(table(merge_mapman_pgsc$NAME)) # generate count table per name

pander(head(merge_mapman_pgsc_table))
pander(head(merge_mapman_pgsc_table2))
dim(merge_mapman_pgsc_table)
dim(merge_mapman_pgsc_table2)

nrow(merge_mapman_pgsc_table2) - length(which(merge_mapman_pgsc_table2$Freq==0))
nrow(merge_mapman_pgsc_table) - length(which(merge_mapman_pgsc_table$Freq==0))

# only mapman bins with more than 1 count
merge_mapman_pgsc_table_unique <- merge_mapman_pgsc_table[which(merge_mapman_pgsc_table$Freq!=0),]
dim(merge_mapman_pgsc_table_unique)

# frequency of mapman bins 
hist(log10(merge_mapman_pgsc_table_unique$Freq), breaks=100, col="grey", main="")
axis(side=3, at=c(0,0.698,1,2,3,4), labels=c(1,5,10,100,1000,10000))
abline(v=c(0,0.698,1,2,3,4))

merge_mapman_pgsc_unique_bincode <- unique(merge_mapman_pgsc$BINCODE)
merge_mapman_pgsc_unique_name <- unique(merge_mapman_pgsc$NAME)
head(merge_mapman_pgsc_unique_name)
```

## DGE analysis by edgeR
### GREENHOUSE: Control vs. Drought Stress
#### Execute edgeR procedure to identify differentially expresses genes (DGE) comparing control and drought stress conditions for greenhouse samples
\textbf{Use FDR < 0.05}
```{r edgeR GREENHOUSE control vs. stress, echo=TRUE}
# rownumbers of selected samples
group_idx <- which(samplelist_modified[, "cultivation"]=="greenhouse")
# contrasting group factor
group_factor <- as.factor(subset(samplelist_modified, cultivation == "greenhouse")$condition)

# results of edgeR analysis
res_greenhouse <- func_edgeR_dge(counts_keep = counts_keep,
                            group_idx = group_idx, 
                            group_1 = group_factor,
                            FDR_threshold = 0.05)
#res_greenhouse$sums

# export significant/all results
write.table(res_greenhouse$top_sig, "output/edgeR_res_greenhouse_top_sig.txt", sep="\t")
write.table(res_greenhouse$top_all, "output/edgeR_res_greenhouse_top_all.txt", sep="\t")

# merge signifcant results (DGE) with MapMan annotation
res_greenhouse_sig_up_mapman <- merge(res_greenhouse$top_sig_up, merge_mapman_pgsc_dmg, by.x="row.names", by.y="pgsc_dmg")
res_greenhouse_sig_down_mapman <- merge(res_greenhouse$top_sig_down, merge_mapman_pgsc_dmg, by.x="row.names", by.y="pgsc_dmg")

# export significant results merged with MapMan annotation
write.table(res_greenhouse_sig_up_mapman, "output/res_greenhouse_sig_up_mapman.txt", sep="\t", row.names=F, quote=F)
write.table(res_greenhouse_sig_down_mapman, "output/res_greenhouse_sig_down_mapman.txt", sep="\t", row.names=F, quote=F)
```


### FIELD: Control vs. Drought Stress
```{r edgeR FIELD control vs. stress, echo=TRUE}
group_idx <- which(samplelist_modified[, "cultivation"]=="field")
group_factor <- as.factor(subset(samplelist_modified, cultivation == "field")$condition)
#res_field <- func_edgeR_dge(group_idx, group_1 = group_factor)

res_field <- func_edgeR_dge(counts_keep = counts_keep,
                            group_idx = group_idx, 
                            group_1 = group_factor,
                            FDR_threshold = 0.05)
#res_field$sums

write.table(res_field$top_sig, "output/edgeR_res_field_top_sig.txt", sep="\t")
write.table(res_field$top_all, "output/edgeR_res_field_top_all.txt", sep="\t")

res_field_sig_up_mapman <- merge(res_field$top_sig_up, merge_mapman_pgsc_dmg, by.x="row.names", by.y="pgsc_dmg")
res_field_sig_down_mapman <- merge(res_field$top_sig_down, merge_mapman_pgsc_dmg, by.x="row.names", by.y="pgsc_dmg")

write.table(res_field_sig_up_mapman, "output/res_field_sig_up_mapman.txt", sep="\t", row.names=F, quote=F)
write.table(res_field_sig_down_mapman, "output/res_field_sig_down_mapman.txt", sep="\t", row.names=F, quote=F)
```


### Venn Diagrams
Venn diagram of differentially expressed genes for control vs. stress comparing all samples either from greenhouse or field
```{r venn diagram greenhouse-field control vs. stress ALL}
venn.plot=func_venn_diagram_2(rownames(res_greenhouse$top_sig), 
                              rownames(res_field$top_sig), 
                              lab1="Greenhouse", lab2="Field", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```



```{r venn diagram greenhouse-field control vs. stress DOWN}
venn.plot=func_venn_diagram_2(rownames(res_greenhouse$top_sig_down), 
                              rownames(res_field$top_sig_down), 
                              lab1="Greenhouse", lab2="Field", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```


```{r intersection greenhouse-field control vs. stress DOWN}
intersect_cultivation_condition_down <- intersect(rownames(res_field$top_sig_down), 
                                                  rownames(res_greenhouse$top_sig_down))

intersect_cultivation_condition_down_res <- cbind(res_field$top_sig_down[rownames(res_field$top_sig_down) %in% intersect_cultivation_condition_down,], res_greenhouse$top_sig_down[rownames(res_greenhouse$top_sig_down) %in% intersect_cultivation_condition_down,])

colnames(intersect_cultivation_condition_down_res) <- c("logFC_field","logCPM_field","PValue_field","FDR_field",
                                                        "logFC_gh","logCPM_gh","PValue_gh","FDR_gh")

intersect_cultivation_condition_down_res$logFC_mean <- apply(cbind(intersect_cultivation_condition_down_res$logFC_field,
                                                                   intersect_cultivation_condition_down_res$logFC_gh),1,mean)
intersect_cultivation_condition_down_res$logFC_sum <- apply(cbind(intersect_cultivation_condition_down_res$logFC_field,
                                                                  intersect_cultivation_condition_down_res$logFC_gh),1,sum)

intersect_cultivation_condition_down_res <- merge(intersect_cultivation_condition_down_res, 
                                                  merge_mapman_pgsc_dmg, by.x="row.names", by.y="pgsc_dmg")

intersect_cultivation_condition_down_res <- intersect_cultivation_condition_down_res[order(intersect_cultivation_condition_down_res$logFC_sum),]

pander(head(intersect_cultivation_condition_down_res[,c(1,2,6,10:13)], n=25))

write.table(intersect_cultivation_condition_down_res, "output/intersect_cultivation_condition_down_res.txt", 
            sep="\t", row.names=F, quote=F)
```



```{r venn diagram greenhouse-field control vs. stress UP}
venn.plot=func_venn_diagram_2(rownames(res_greenhouse$top_sig_up), 
                              rownames(res_field$top_sig_up), 
                              lab1="Greenhouse", lab2="Field", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```


```{r intersection greenhouse-field control vs. stress UP}
intersect_cultivation_condition_up <- intersect(rownames(res_field$top_sig_up), 
                                                rownames(res_greenhouse$top_sig_up))

intersect_cultivation_condition_up_res <- cbind(res_field$top_sig_up[rownames(res_field$top_sig_up) %in% intersect_cultivation_condition_up,], res_greenhouse$top_sig_up[rownames(res_greenhouse$top_sig_up) %in% intersect_cultivation_condition_up,])

colnames(intersect_cultivation_condition_up_res) <- c("logFC_field","logCPM_field","PValue_field","FDR_field",
                                                        "logFC_gh","logCPM_gh","PValue_gh","FDR_gh")
intersect_cultivation_condition_up_res$logFC_mean <- apply(cbind(intersect_cultivation_condition_up_res$logFC_field,intersect_cultivation_condition_up_res$logFC_gh),1,mean)
intersect_cultivation_condition_up_res$logFC_sum <- apply(cbind(intersect_cultivation_condition_up_res$logFC_field,intersect_cultivation_condition_up_res$logFC_gh),1,sum)

intersect_cultivation_condition_up_res <- merge(intersect_cultivation_condition_up_res, merge_mapman_pgsc_dmg, by.x="row.names", by.y="pgsc_dmg")

intersect_cultivation_condition_up_res <- intersect_cultivation_condition_up_res[order(intersect_cultivation_condition_up_res$logFC_sum, decreasing=T),]

pander(head(intersect_cultivation_condition_up_res[,c(1,2,6,10:13)], n=20))
write.table(intersect_cultivation_condition_up_res, "output/intersect_cultivation_condition_up_res.txt", sep="\t", row.names=F, quote=F)
```


## Execute 2nd edgeR procedure function for SUBSETS
### GREENHOUSE + CONTROL: Tolerant vs. Sensitive
Execute edgeR procedure to identify differentially expresses genes (DGE) comparing tolerant and sensitive groups for greenhouse AND control samples
```{r edgeR greenhouse control - tolerant vs. sensitive, echo=TRUE}
group_idx <- which(samplelist_greenhouse[, "condition"]=="control")
group_factor <- as.factor(subset(samplelist_greenhouse, condition == "control")$tolerance)
res_greenhouse_control <- func_edgeR_dge(counts_keep_greenhouse, group_idx, group_1 = group_factor)
#res_greenhouse_control$sums
```


### GREENHOUSE + STRESS: Tolerant vs. Sensitive
Execute edgeR procedure to identify differentially expresses genes (DGE) comparing tolerant and sensitive groups for greenhouse AND stress samples
** pos(1) means HIGHER expression in tolerant than in sensitive cultivars**
** neg(-1) means LOWER expression in tolerant than in sensitive cultivars**
```{r edgeR greenhouse stress - tolerant vs. sensitive, echo=TRUE}
group_idx <- which(samplelist_greenhouse[, "condition"]=="drought stress")
group_factor <- as.factor(subset(samplelist_greenhouse, condition == "drought stress")$tolerance)
res_greenhouse_stress <- func_edgeR_dge(counts_keep_greenhouse, group_idx, group_1 = group_factor)
# res_greenhouse_stress$sums
```

#### Venn Diagrams
**Venn diagram of differentially expressed genes for sensitive vs. tolerant comparing control and stress samples from greenhouse**
```{r venn diagram greenhouse-control-stress - tolerant vs. sensitive}
venn.plot=func_venn_diagram_2(rownames(res_greenhouse_control$top_sig), 
                              rownames(res_greenhouse_stress$top_sig), 
                              lab1="Control", lab2="Stress", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

Positive logFC which means **HIGHER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse-control-stress - tolerant vs. sensitive pos}
venn.plot=func_venn_diagram_2(rownames(res_greenhouse_control$top_sig_up), 
                              rownames(res_greenhouse_stress$top_sig_up), 
                              lab1="Control", lab2="Stress", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

Negative logFC which means **LOWER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse-control-stress - tolerant vs. sensitive neg}
venn.plot=func_venn_diagram_2(rownames(res_greenhouse_control$top_sig_down), 
                              rownames(res_greenhouse_stress$top_sig_down), 
                              lab1="Control", lab2="Stress", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```


### FIELD + CONTROL: Tolerant vs. Sensitive
Execute edgeR procedure to identify differentially expresses genes (DGE) comparing tolerant and sensitive groups for field AND control samples
```{r edgeR field-control - tolerant vs. sensitive, echo=TRUE}
group_idx <- which(samplelist_field[, "condition"]=="control")
group_factor <- as.factor(subset(samplelist_field, condition == "control")$tolerance)
res_field_control <- func_edgeR_dge(counts_keep_field, group_idx, group_1 = group_factor)
# res_field_control$sums
```


### FIELD + STRESS: Tolerant vs. Sensitive
Execute edgeR procedure to identify differentially expresses genes (DGE) comparing tolerant and sensitive groups for field AND stress samples
```{r edgeR field-stress - tolerant vs. sensitive, echo=TRUE}
group_idx <- which(samplelist_field[, "condition"]=="drought stress")
group_factor <- as.factor(subset(samplelist_field, condition == "drought stress")$tolerance)
res_field_stress <- func_edgeR_dge(counts_keep_field, group_idx, group_1 = group_factor)
# res_field_stress$sums
```


#### Venn Diagrams
**Venn diagram of differentially expressed genes for sensitive vs. tolerant comparing control and stress samples from field**
```{r venn diagram field-control-stress - tolerant vs. sensitive}
venn.plot=func_venn_diagram_2(rownames(res_field_control$top_sig), 
                              rownames(res_field_stress$top_sig), 
                              lab1="Control", lab2="Stress", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

Positive logFC which means **HIGHER** expression in tolerant than in sensitive cultivars
```{r venn diagram field-control-stress - tolerant vs. sensitive pos}
venn.plot=func_venn_diagram_2(rownames(res_field_control$top_sig_up), 
                              rownames(res_field_stress$top_sig_up), 
                              lab1="Control", lab2="Stress", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

Negative logFC which means **LOWER** expression in tolerant than in sensitive cultivars
```{r venn diagram field-control-stress - tolerant vs. sensitive neg}
venn.plot=func_venn_diagram_2(rownames(res_field_control$top_sig_down), 
                              rownames(res_field_stress$top_sig_down), 
                              lab1="Control", lab2="Stress", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

**Venn diagram of differentially expressed genes for sensitive vs. tolerant comparing samples from field and greenhouse**
**CONTROL SAMPLES**
Positive logFC which means **HIGHER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse-field-control - tolerant vs. sensitive pos}
venn.plot=func_venn_diagram_2(rownames(res_field_control$top_sig_up), 
                              rownames(res_greenhouse_control$top_sig_up), 
                              lab1="Field", lab2="Greenhouse", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

**CONTROL SAMPLES**
Negative logFC which means **LOWER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse-field-control - tolerant vs. sensitive neg}
venn.plot=func_venn_diagram_2(rownames(res_field_control$top_sig_down), 
                              rownames(res_greenhouse_control$top_sig_down), 
                              lab1="Field", lab2="Greenhouse", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

**DROUGHT STRESS SAMPLES**
Positive logFC which means **HIGHER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse-field-stress - tolerant vs. sensitive pos}
venn.plot=func_venn_diagram_2(rownames(res_field_stress$top_sig_up), 
                              rownames(res_greenhouse_stress$top_sig_up), 
                              lab1="Field", lab2="Greenhouse", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

**DROUGHT STRESS SAMPLES**
Negative logFC which means **LOWER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse-field-stress - tolerant vs. sensitive neg}
venn.plot=func_venn_diagram_2(rownames(res_field_stress$top_sig_down), 
                              rownames(res_greenhouse_stress$top_sig_down), 
                              lab1="Field", lab2="Greenhouse", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

**Compare 4 different DGE**
Positive logFC which means **HIGHER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse/field control/stress - tolerant vs. sensitive pos}
venn.plot=func_venn_diagram_4(rownames(res_field_control$top_sig_up), 
                              rownames(res_field_stress$top_sig_up),
                              rownames(res_greenhouse_control$top_sig_up),
                              rownames(res_greenhouse_stress$top_sig_up),
                              lab1 = "Field \n Control", 
                              lab2 = "Field \n Stress",
                              lab3 = "Greenhouse \n Control", 
                              lab4 = "Greenhouse \n Stress", 
                              filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

Negative logFC which means **LOWER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse/field control/stress - tolerant vs. sensitive neg}
venn.plot=func_venn_diagram_4(rownames(res_field_control$top_sig_down), 
                              rownames(res_field_stress$top_sig_down),
                              rownames(res_greenhouse_control$top_sig_down),
                              rownames(res_greenhouse_stress$top_sig_down),
                              lab1 = "Field \n Control", 
                              lab2 = "Field \n Stress",
                              lab3 = "Greenhouse \n Control", 
                              lab4 = "Greenhouse \n Stress", 
                              filepath=NULL)
plot.new()
grid.draw(venn.plot)
```


### Save tables for PageMan
```{r save tables for PageMan tolerant vs. sensitive}
res_greenhouse_control_pageman <- func_pageman_res(res_greenhouse_control)
res_greenhouse_stress_pageman <- func_pageman_res(res_greenhouse_stress)
res_field_control_pageman <- func_pageman_res(res_field_control)
res_field_stress_pageman <- func_pageman_res(res_field_stress)

write.table(res_greenhouse_control_pageman, "output/pageman_greenhouse_control_all.txt", 
            sep="\t", row.names=F, quote=F)
write.table(res_greenhouse_stress_pageman, "output/pageman_greenhouse_stress_all.txt", 
            sep="\t", row.names=F, quote=F)
write.table(res_field_control_pageman, "output/pageman_field_control_all.txt", 
            sep="\t", row.names=F, quote=F)
write.table(res_field_stress_pageman, "output/pageman_field_stress_all.txt", 
            sep="\t", row.names=F, quote=F)
```


```{r PageMan tolerant vs. sensitive}
# merge to mapman bins
# dir ==  1 --> tol > sens
# dir == -1 --> sens > tol
res_field_control_pageman_mapman_T <- merge(res_field_control_pageman[which(res_field_control_pageman$dir == 1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
res_field_control_pageman_mapman_S <- merge(res_field_control_pageman[which(res_field_control_pageman$dir == -1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
res_field_stress_pageman_mapman_T <- merge(res_field_stress_pageman[which(res_field_stress_pageman$dir == 1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
res_field_stress_pageman_mapman_S <- merge(res_field_stress_pageman[which(res_field_stress_pageman$dir == -1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
# greenhouse
res_greenhouse_control_pageman_mapman_T <- merge(res_greenhouse_control_pageman[which(res_greenhouse_control_pageman$dir == 1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
res_greenhouse_control_pageman_mapman_S <- merge(res_greenhouse_control_pageman[which(res_greenhouse_control_pageman$dir == -1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
res_greenhouse_stress_pageman_mapman_T <- merge(res_greenhouse_stress_pageman[which(res_greenhouse_stress_pageman$dir == 1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
res_greenhouse_stress_pageman_mapman_S <- merge(res_greenhouse_stress_pageman[which(res_greenhouse_stress_pageman$dir == -1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")

# res_field_control_pageman_mapman_T <- res_field_control_pageman_mapman_T[order(res_field_control_pageman_mapman_T$BINCODE),]
# res_field_control_pageman_mapman_S <- res_field_control_pageman_mapman_S[order(res_field_control_pageman_mapman_S$BINCODE),]
```


```{r PageMan tolerant vs. sensitive --> cytochrome P450}
# number of significant genes in bin 26.10*
# field
length(unique(subset(res_field_control_pageman_mapman_T$logFC, 
              grepl("26.10*",res_field_control_pageman_mapman_T$BINCODE))))
length(unique(subset(res_field_control_pageman_mapman_S$logFC, 
              grepl("26.10*",res_field_control_pageman_mapman_S$BINCODE))))
length(unique(subset(res_field_stress_pageman_mapman_T$logFC, 
              grepl("26.10*",res_field_stress_pageman_mapman_T$BINCODE))))
length(unique(subset(res_field_stress_pageman_mapman_S$logFC, 
              grepl("26.10*",res_field_stress_pageman_mapman_S$BINCODE))))
# greenhouse
length(unique(subset(res_greenhouse_control_pageman_mapman_T$logFC, 
              grepl("26.10*",res_greenhouse_control_pageman_mapman_T$BINCODE))))
length(unique(subset(res_greenhouse_control_pageman_mapman_S$logFC, 
              grepl("26.10*",res_greenhouse_control_pageman_mapman_S$BINCODE))))
length(unique(subset(res_greenhouse_stress_pageman_mapman_T$logFC, 
              grepl("26.10*",res_greenhouse_stress_pageman_mapman_T$BINCODE))))
length(unique(subset(res_greenhouse_stress_pageman_mapman_S$logFC, 
              grepl("26.10*",res_greenhouse_stress_pageman_mapman_S$BINCODE))))

#intersect(res_field_control_pageman_mapman_T$pgsc_dmp[which(grepl("26.10*",res_field_control_pageman_mapman_T$BINCODE))],
#          res_field_control_pageman_mapman_S$pgsc_dmp[which(grepl("26.10*",res_field_control_pageman_mapman_S$BINCODE))])
```


```{r PageMan tolerant vs. sensitive --> receptor kinases}
# number of significant genes in bin 30.2*
# field
length(unique(subset(res_field_control_pageman_mapman_T$logFC, 
              grepl("30.2*",res_field_control_pageman_mapman_T$BINCODE))))
length(unique(subset(res_field_control_pageman_mapman_S$logFC, 
              grepl("30.2*",res_field_control_pageman_mapman_S$BINCODE))))
length(unique(subset(res_field_stress_pageman_mapman_T$logFC, 
              grepl("30.2*",res_field_stress_pageman_mapman_T$BINCODE))))
length(unique(subset(res_field_stress_pageman_mapman_S$logFC, 
              grepl("30.2*",res_field_stress_pageman_mapman_S$BINCODE))))
# greenhouse
length(unique(subset(res_greenhouse_control_pageman_mapman_T$logFC, 
              grepl("30.2*",res_greenhouse_control_pageman_mapman_T$BINCODE))))
length(unique(subset(res_greenhouse_control_pageman_mapman_S$logFC, 
              grepl("30.2*",res_greenhouse_control_pageman_mapman_S$BINCODE))))
length(unique(subset(res_greenhouse_stress_pageman_mapman_T$logFC, 
              grepl("30.2*",res_greenhouse_stress_pageman_mapman_T$BINCODE))))
length(unique(subset(res_greenhouse_stress_pageman_mapman_S$logFC, 
              grepl("30.2*",res_greenhouse_stress_pageman_mapman_S$BINCODE))))
```


```{r only receptor kinases (30.2)}
length(which(grepl("30.2*",merge_mapman_pgsc$BINCODE)))
mapman_kinases <- merge_mapman_pgsc[which(grepl("30.2*",merge_mapman_pgsc$BINCODE)),]
                               
res_field_control_kinases <- func_merge_res_stress_bin(res_field_control$top_all, mapman_kinases)
res_field_stress_kinases <- func_merge_res_stress_bin(res_field_stress$top_all, mapman_kinases)
res_greenhouse_control_kinases <- func_merge_res_stress_bin(res_greenhouse_control$top_all, mapman_kinases)
res_greenhouse_stress_kinases <- func_merge_res_stress_bin(res_greenhouse_stress$top_all, mapman_kinases)

fdr_kinases <- cbind(field_control = res_field_control_kinases$FDR,
                     field_stress = res_field_stress_kinases$FDR,
                     greenhouse_control = res_greenhouse_control_kinases$FDR,
                     greenhouse_stress = res_greenhouse_stress_kinases$FDR)
fdr_kinases_min <- apply(fdr_kinases, 1, min)
length(which(fdr_kinases_min<0.001))

logfc_kinases <- cbind(field_control = res_field_control_kinases$logFC,
                     field_stress = res_field_stress_kinases$logFC,
                     greenhouse_control = res_greenhouse_control_kinases$logFC,
                     greenhouse_stress = res_greenhouse_stress_kinases$logFC)

rownames(logfc_kinases) <- res_field_stress_kinases$pgsc_dmg
logfc_kinases_max <- abs(apply(logfc_kinases, 1, max))
logfc_kinases_part <- logfc_kinases[which(fdr_kinases_min < 0.001),]
#logfc_kinases_part <- logfc_kinases[which(logfc_kinases_max > 1),]
hist(logfc_kinases_part)
logfc_kinases_part <- unique(merge(logfc_kinases_part, assoc_pgsc[,c(1,5)], by.x="row.names", by.y="pgsc_dmg"))
dim(logfc_kinases_part)
```


```{r heatmap only receptor kinases bin}
mat_data <- as.matrix(logfc_kinases_part[,c(2:5)])
rownames(mat_data) <- logfc_kinases_part[,6]
distance = dist(mat_data, method = "euclidian")
cluster = hclust(distance, method = "average")
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 299)
col_breaks = c(seq(-4,-1,length=100), # for blue
               seq(-1,1,length=100), # for white
               seq(1,4,length=100)) # for red

pdf("figures/heatmap_log2FC_kinases.pdf", width=6, height=15)

lmat=rbind(4:3, 2:1)
lhei=c(0.4, 5)
lwid=c(1, 4)

layout(mat = lmat, widths = lwid, heights = lhei)

heatmap.2(mat_data,
  #cellnote = signif2,  # same data set for cell labels
  #notecex=1.5, # numeric scaling factor for cellnote items
  colsep=1:ncol(mat_data), 
  rowsep=1:nrow(mat_data),
  sepwidth=c(0.01,0.05),
  sepcolor="black",
  main = "", # heat map title
  #notecol="black",      # change font color of cell labels to black
  density.info="none",  # turns off density plot inside color legend
  trace="none",         # turns off trace lines inside the heat map
  margins =c(8,18),     # widens margins around plot
  keysize=1.5,
  cexRow=1.2,
  cexCol=1.5,
  adjCol=c(0,1),
  srtCol = 330,
  lhei=c(0.4, 5),
  col=my_palette,       # use on color palette defined earlier 
  breaks=col_breaks,    # enable color transition at specified limits
  #RowSideColors=mycolor,
  #ColSideColors=column_color,
  Rowv=as.dendrogram(cluster),
  dendrogram="row",     # only draw a row dendrogram
  Colv="NA")            # turn off column clustering```

dev.off()
@
```





## save workspace
```{r save workspace}
save.image("exp_counts_edgeR.RData")
```

