---
title: "Expected Counts Data Analyis - edgeR"
author: "Heike Sprenger"
date: "Tuesday, May 05, 2015"
output:
  html_document:
    highlight: tango
    number_section: yes
    theme: cerulean
    toc: yes
    toc_depth: 4
---

# Input: Expected counts per gene (from RSEM)

# Set working directory  
```{r set working directory}
#getwd()
#setwd("D:/work/repos/trost_transcriptomics")
#setwd("~/work/repos/trost_transcriptomics")
```

[solution for issue with working directory and knitr](https://github.com/yihui/knitr/issues/277)

# Load workspace, packages and scripts
```{r load workspace, message=FALSE}
# load packages
library(knitr)
library(pcaMethods)
library(edgeR)
library(VennDiagram)
library(gplots)
library(RColorBrewer)
library(plyr)
library(reshape2)
library(pander)

# set options for pander
panderOptions('table.split.table', 200)

# set options for knitr
opts_chunk$set(fig.width=5, fig.height=5, cache=FALSE, highlight = TRUE, fig.show="asis")
opts_knit$set(root.dir = '../')

# load workspace
#load("exp_counts_edgeR.RData")
```


# Samplelist processing by using `Modify_RNASeq_Samplelist.Rmd`
```{r child = 'Modify_RNASeq_Samplelist.Rmd'}
# knit('scripts/Modify_RNASeq_Samplelist.Rmd')
```


# Source R functions
```{r source R functions}
source("../libpurzel/func_venn_diagram.R")
source("../libpurzel/func_edgeR_dge.R")
source("../libpurzel/func_pageman_res.R")
source("../libpurzel/func_merge_sig_res_mapman.R")
source("../libpurzel/func_edgeR_top.R")
source("../libpurzel/func_intersecting_dge.R")
source("../libpurzel/func_final_DGE_list.R")
source("../libpurzel/func_order_by_MapMan_BinCode.R")
source("../libpurzel/func_create_type.R")
source("../libpurzel/func_combine_two_vectors.R")
source("../libpurzel/func_combine_two_final_DGE_lists.R")
```


# Load count data and additional information
## Load filtered expected counts
```{r load filtered expected conunts}
counts_keep <- read.table("output/counts_keep.txt", sep="\t", header=T)

# rownames of filtered expected counts (DMG identifier)
#rownames_intersect_greenhouse_field <- rownames(counts_keep)
#length(rownames_intersect_greenhouse_field)

# greenhouse
counts_keep_greenhouse <- counts_keep[,which(samplelist_ordered$cultivation=="greenhouse")]

# field
counts_keep_field <- counts_keep[,which(samplelist_ordered$cultivation=="field")]
```


## Load annotation files
```{r load annotation files}
# PGSC annotation table with gene, transcript, protein IDs and functional annotation
assoc_pgsc <- read.table("data/PGSC_DM_v3.4_g2t2c2p2func_edit.txt")
colnames(assoc_pgsc) <- c("pgsc_dmg", "pgsc_dmt", "pgsc_dmc", "pgsc_dmp", "func")
pander(head(assoc_pgsc))

# only unique entries for DMG and functional annotation
# assoc_pgsc_dmg <- unique(assoc_pgsc[,c(1,5)])
assoc_pgsc_unique_dmg <- unique(assoc_pgsc$pgsc_dmg)
assoc_pgsc_unique_dmt <- unique(assoc_pgsc$pgsc_dmt)
assoc_pgsc_unique_dmp <- unique(assoc_pgsc$pgsc_dmp)

# table with MapMan annotation per protein ID (pgsc_dmp)
assoc_mapman <- read.table("data/mapman_cleaned_output_new.txt", header = T)
dim(assoc_mapman)
colnames(assoc_mapman)

# use only mapman entries where the TYPE is TRUE
assoc_mapman_true <- droplevels(subset(assoc_mapman, assoc_mapman$TYPE==TRUE))
dim(assoc_mapman_true)

# only unique entries for DMP from assoc_mapman
assoc_mapman_unique_ids <- unique(assoc_mapman_true$IDENTIFIER)
length(assoc_mapman_unique_ids)
length(assoc_pgsc_unique_dmp)
# complete overlap with assoc_pgsc_unique_dmp: 56218 identifier!

# merge PGSC annotation table with MapMan table (via DMP identifier)
merge_mapman_pgsc <- merge(assoc_pgsc, assoc_mapman_true, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
colnames(merge_mapman_pgsc)
dim(merge_mapman_pgsc)
write.table(merge_mapman_pgsc, "data/merge_mapman_pgsc.txt", sep="\t")

# unique MapMan classifications per gene
merge_mapman_pgsc_dmg <- unique(merge_mapman_pgsc[,c(2,5:7)])
pander(head(merge_mapman_pgsc_dmg))

merge_mapman_pgsc_dmp <- unique(merge_mapman_pgsc[,c(1,2,5:7)])

# count how many different bincodes were associated per gene identifier
table(count(merge_mapman_pgsc_dmg, vars = "pgsc_dmg")$freq)

# count how many different bincodes were associated per protein identifier
table(count(merge_mapman_pgsc, vars = "pgsc_dmp")$freq)
```


### Count how many different protein identifier were associated per bincode
```{r count how many different protein identifier were associated per bincode}
mapman_name <- assoc_mapman_true$NAME
mapman_bincode <- assoc_mapman_true$BINCODE
mapman_bincode_table <- table(mapman_bincode)

mapman_bincode_order <- cbind(BINCODE = as.character(mapman_bincode), 
                              NAME = as.character(mapman_name),
                              order = 1:57441)

# unique mapman bincode and name (without order column)
mapman_bincode_order_unique <- as.data.frame( unique(mapman_bincode_order[,-3]) )
head(mapman_bincode_order_unique)

# sort unique mapman table by bincodes to get index (later needed for order)
mapman_bincode_order_unique_alphabetical <- mapman_bincode_order_unique[order(mapman_bincode_order_unique$BINCODE),]
mapman_bincode_order_unique_index <- as.numeric(row.names(mapman_bincode_order_unique_alphabetical))

# merge unique mapman table with count table per 
mapman_bincode_merge <- merge(mapman_bincode_order_unique, mapman_bincode_table, by.x = "BINCODE", by.y = "mapman_bincode")
mapman_bincode_merge_ordered <- mapman_bincode_merge[order(mapman_bincode_order_unique_index),]
head(mapman_bincode_merge_ordered)

write.table(mapman_bincode_merge_ordered, "output/mapman_bincode_frequency_per_dmp.txt", sep="\t", row.names = F, quote = F)
```


## Melt MapMan table
```{r melt mapman table}
# melt
melt_mapman_pgsc <- melt(merge_mapman_pgsc, id=c("BINCODE")) 
pander(head(melt_mapman_pgsc))

# cast
cast_mapman_pgsc <- dcast(merge_mapman_pgsc, merge_mapman_pgsc$NAME ~ merge_mapman_pgsc$pgsc_dmp) 
dim(cast_mapman_pgsc)

# generate count table
merge_mapman_pgsc_table <- as.data.frame(table(merge_mapman_pgsc$BINCODE)) # generate count table per bincode
merge_mapman_pgsc_table2 <- as.data.frame(table(merge_mapman_pgsc$NAME)) # generate count table per name

pander(head(merge_mapman_pgsc_table))
pander(head(merge_mapman_pgsc_table2))
dim(merge_mapman_pgsc_table)
dim(merge_mapman_pgsc_table2)

nrow(merge_mapman_pgsc_table2) - length(which(merge_mapman_pgsc_table2$Freq==0))
nrow(merge_mapman_pgsc_table) - length(which(merge_mapman_pgsc_table$Freq==0))

# only mapman bins with more than 1 count
merge_mapman_pgsc_table_unique <- merge_mapman_pgsc_table[which(merge_mapman_pgsc_table$Freq!=0),]
dim(merge_mapman_pgsc_table_unique)

# frequency of mapman bins 
hist(log10(merge_mapman_pgsc_table_unique$Freq), breaks=100, col="grey", main="")
axis(side=3, at=c(0,0.698,1,2,3,4), labels=c(1,5,10,100,1000,10000))
abline(v=c(0,0.698,1,2,3,4))

merge_mapman_pgsc_unique_bincode <- unique(merge_mapman_pgsc$BINCODE)
merge_mapman_pgsc_unique_name <- unique(merge_mapman_pgsc$NAME)
head(merge_mapman_pgsc_unique_name)
```

# DGE analysis by edgeR
## Control vs. Drought Stress
### GREENHOUSE: Control vs. Drought Stress
**Execute edgeR procedure to identify differentially expresses genes (DGE) comparing control and drought stress conditions for greenhouse samples**
\textbf{Use FDR < 0.05}
```{r edgeR GREENHOUSE control vs. stress, echo=TRUE}
# rownumbers of selected samples
group_idx <- which(samplelist_modified[, "cultivation"]=="greenhouse")
# contrasting group factor
group_factor <- as.factor(subset(samplelist_modified, cultivation == "greenhouse")$condition)

# results of edgeR analysis
res_greenhouse <- func_edgeR_dge(counts_keep = counts_keep,
                            group_idx = group_idx, 
                            group_1 = group_factor,
                            FDR_threshold = 0.05)
#res_greenhouse$sums

# export significant/all results
write.table(res_greenhouse$top_sig, "output/edgeR_res_greenhouse_top_sig.txt", sep="\t")
write.table(res_greenhouse$top_all, "output/edgeR_res_greenhouse_top_all.txt", sep="\t")

# merge signifcant results (DGE) with MapMan annotation
res_greenhouse_sig_up_mapman <- merge(res_greenhouse$top_sig_up, merge_mapman_pgsc_dmg, by.x="row.names", by.y="pgsc_dmg")
res_greenhouse_sig_down_mapman <- merge(res_greenhouse$top_sig_down, merge_mapman_pgsc_dmg, by.x="row.names", by.y="pgsc_dmg")

# export significant results merged with MapMan annotation
write.table(res_greenhouse_sig_up_mapman, "output/res_greenhouse_sig_up_mapman.txt", sep="\t", row.names=F, quote=F)
write.table(res_greenhouse_sig_down_mapman, "output/res_greenhouse_sig_down_mapman.txt", sep="\t", row.names=F, quote=F)
```


### FIELD: Control vs. Drought Stress
```{r edgeR FIELD control vs. stress, echo=TRUE}
group_idx <- which(samplelist_modified[, "cultivation"]=="field")
group_factor <- as.factor(subset(samplelist_modified, cultivation == "field")$condition)
#res_field <- func_edgeR_dge(group_idx, group_1 = group_factor)

res_field <- func_edgeR_dge(counts_keep = counts_keep,
                            group_idx = group_idx, 
                            group_1 = group_factor,
                            FDR_threshold = 0.05)
#res_field$sums

write.table(res_field$top_sig, "output/edgeR_res_field_top_sig.txt", sep="\t")
write.table(res_field$top_all, "output/edgeR_res_field_top_all.txt", sep="\t")

res_field_sig_up_mapman <- merge(res_field$top_sig_up, merge_mapman_pgsc_dmg, by.x="row.names", by.y="pgsc_dmg")
res_field_sig_down_mapman <- merge(res_field$top_sig_down, merge_mapman_pgsc_dmg, by.x="row.names", by.y="pgsc_dmg")

write.table(res_field_sig_up_mapman, "output/res_field_sig_up_mapman.txt", sep="\t", row.names=F, quote=F)
write.table(res_field_sig_down_mapman, "output/res_field_sig_down_mapman.txt", sep="\t", row.names=F, quote=F)
```


### Venn Diagrams
#### All differentially experessed genes
Venn diagram of differentially expressed genes for control vs. stress comparing all samples either from greenhouse or field
```{r venn diagram greenhouse-field control vs. stress ALL}
venn.plot <- func_venn_diagram_2(rownames(res_greenhouse$top_sig), 
                              rownames(res_field$top_sig), 
                              lab1="Greenhouse", lab2="Field", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```


#### Only Down-regulated genes
```{r venn diagram greenhouse-field control vs. stress DOWN}
venn.plot <- func_venn_diagram_2(rownames(res_greenhouse$top_sig_down), 
                              rownames(res_field$top_sig_down), 
                              lab1="Greenhouse", lab2="Field", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```


#### Intersection greenhouse-field control vs. stress DOWN
```{r intersection greenhouse-field control vs. stress DOWN}
# return_complete == TRUE means that intersect_res will be merged with COMPLETE MAPMAN_PGSC annotation file
intersect_cultivation_condition_down_res_complete <- 
  func_intersecting_dge(set1 = res_field$top_sig_down, 
                        set2 = res_greenhouse$top_sig_down, 
                        lab1 = "field",
                        lab2 = "gh",
                        return_complete = TRUE)

# return_complete == FALSE means that intersect_res will be merged with MAPMAN_PGSC annotation file containing only UNIQUE DMG ids
intersect_cultivation_condition_down_res <- 
  func_intersecting_dge(set1 = res_field$top_sig_down, 
                        set2 = res_greenhouse$top_sig_down, 
                        lab1 = "field",
                        lab2 = "gh",
                        return_complete = FALSE)

# first 25 intersecting DGE (columns: row.names, logFC of set1, logFC of set2, logFC mean, logFC sum, annotations)
pander(head(  intersect_cultivation_condition_down_res[,c(1,2,6,10:13)], n=25))

write.table(intersect_cultivation_condition_down_res, "output/intersect_cultivation_condition_down_res.txt", 
            sep="\t", row.names=F, quote=F)
```


#### Only Up-regulated genes
```{r venn diagram greenhouse-field control vs. stress UP}
venn.plot <- func_venn_diagram_2(rownames(res_greenhouse$top_sig_up), 
                              rownames(res_field$top_sig_up), 
                              lab1="Greenhouse", lab2="Field", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```


#### Intersection greenhouse-field control vs. stress UP
```{r intersection greenhouse-field control vs. stress UP}
# return_complete == TRUE means that intersect_res will be merged with COMPLETE MAPMAN_PGSC annotation file
intersect_cultivation_condition_up_res_complete <- 
  func_intersecting_dge(set1 = res_field$top_sig_up, 
                        set2 = res_greenhouse$top_sig_up, 
                        lab1 = "field",
                        lab2 = "gh",
                        return_complete = TRUE)

# return_complete == FALSE means that intersect_res will be merged with MAPMAN_PGSC annotation file containing only UNIQUE DMG ids
intersect_cultivation_condition_up_res <- 
  func_intersecting_dge(set1 = res_field$top_sig_up, 
                        set2 = res_greenhouse$top_sig_up, 
                        lab1 = "field",
                        lab2 = "gh",
                        return_complete = FALSE)

# first 25 intersecting DGE (columns: row.names, logFC of set1, logFC of set2, logFC mean, logFC sum, annotations)
pander(head(  intersect_cultivation_condition_up_res[,c(1,2,6,10:13)], n=25))

write.table(intersect_cultivation_condition_up_res, 
            "output/intersect_cultivation_condition_up_res.txt", 
            sep="\t", row.names=F, quote=F)
```


# Execute 2nd edgeR procedure function for SUBSETS
## Tolerant vs. Sensitive
### GREENHOUSE + CONTROL: Tolerant vs. Sensitive
Execute edgeR procedure to identify differentially expresses genes (DGE) comparing tolerant and sensitive groups for greenhouse AND control samples
```{r edgeR greenhouse control - tolerant vs. sensitive, echo=TRUE}
group_idx <- which(samplelist_greenhouse[, "condition"] == "control")
group_factor <- as.factor(subset(samplelist_greenhouse, condition == "control")$tolerance)
res_greenhouse_control <- func_edgeR_dge(counts_keep_greenhouse, group_idx, group_1 = group_factor)
#res_greenhouse_control$sums
```


### GREENHOUSE + STRESS: Tolerant vs. Sensitive
Execute edgeR procedure to identify differentially expresses genes (DGE) comparing tolerant and sensitive groups for greenhouse AND stress samples
**pos(1) means HIGHER expression in tolerant than in sensitive cultivars**
**neg(-1) means LOWER expression in tolerant than in sensitive cultivars**
```{r edgeR greenhouse stress - tolerant vs. sensitive, echo=TRUE}
group_idx <- which(samplelist_greenhouse[, "condition"]=="drought stress")
group_factor <- as.factor(subset(samplelist_greenhouse, condition == "drought stress")$tolerance)
res_greenhouse_stress <- func_edgeR_dge(counts_keep_greenhouse, group_idx, group_1 = group_factor)
# res_greenhouse_stress$sums
```


#### Intersection control-stress greenhouse - sensitive vs. tolerant
```{r intersection control-stress greenhouse - sensitive vs. tolerant}
# # up-regulated genes
# intersect_greenhouse_sens_vs_tol_up <- 
#   func_intersecting_dge(set1 = res_greenhouse_control$top_sig_up, 
#                         set2 = res_greenhouse_stress$top_sig_up, 
#                         lab1 = "control", lab2 = "stress",
#                         return_complete = FALSE)
# 
# intersect_greenhouse_sens_vs_tol_up_complete <- 
#   func_intersecting_dge(set1 = res_greenhouse_control$top_sig_up, 
#                         set2 = res_greenhouse_stress$top_sig_up, 
#                         lab1 = "control", lab2 = "stress",
#                         return_complete = TRUE)
# 
# dim(intersect_greenhouse_sens_vs_tol_up_complete)
# dim(intersect_greenhouse_sens_vs_tol_up)
# 
# # first 25 intersecting DGE (columns: row.names, logFC of set1, logFC of set2, logFC mean, logFC sum, annotations)
# pander(head(  intersect_greenhouse_sens_vs_tol_up[,c(1,2,6,10:13)], n=25))
# 
# write.table(intersect_tolerant_up, "output/intersect_tolerant_up.txt", sep="\t", row.names=F, quote=F)
```


#### Venn Diagrams
**Venn diagram of differentially expressed genes for sensitive vs. tolerant comparing control and stress samples from greenhouse**
```{r venn diagram greenhouse-control-stress - tolerant vs. sensitive}
venn.plot <- func_venn_diagram_2(rownames(res_greenhouse_control$top_sig), 
                              rownames(res_greenhouse_stress$top_sig), 
                              lab1="Control", lab2="Stress", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

Positive logFC which means **HIGHER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse-control-stress - tolerant vs. sensitive pos}
venn.plot <- func_venn_diagram_2(rownames(res_greenhouse_control$top_sig_up), 
                              rownames(res_greenhouse_stress$top_sig_up), 
                              lab1="Control", lab2="Stress", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

Negative logFC which means **LOWER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse-control-stress - tolerant vs. sensitive neg}
venn.plot <- func_venn_diagram_2(rownames(res_greenhouse_control$top_sig_down), 
                              rownames(res_greenhouse_stress$top_sig_down), 
                              lab1="Control", lab2="Stress", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```


### FIELD + CONTROL: Tolerant vs. Sensitive
Execute edgeR procedure to identify differentially expresses genes (DGE) comparing tolerant and sensitive groups for field AND control samples
```{r edgeR field-control - tolerant vs. sensitive, echo=TRUE}
group_idx <- which(samplelist_field[, "condition"]=="control")
group_factor <- as.factor(subset(samplelist_field, condition == "control")$tolerance)
res_field_control <- func_edgeR_dge(counts_keep_field, group_idx, group_1 = group_factor)
# res_field_control$sums
```


### FIELD + STRESS: Tolerant vs. Sensitive
Execute edgeR procedure to identify differentially expresses genes (DGE) comparing tolerant and sensitive groups for field AND stress samples
```{r edgeR field-stress - tolerant vs. sensitive, echo=TRUE}
group_idx <- which(samplelist_field[, "condition"]=="drought stress")
group_factor <- as.factor(subset(samplelist_field, condition == "drought stress")$tolerance)
res_field_stress <- func_edgeR_dge(counts_keep_field, group_idx, group_1 = group_factor)
# res_field_stress$sums
```


#### Venn Diagrams
**Venn diagram of differentially expressed genes for sensitive vs. tolerant comparing control and stress samples from field**
```{r venn diagram field-control-stress - tolerant vs. sensitive}
venn.plot <- func_venn_diagram_2(rownames(res_field_control$top_sig), 
                              rownames(res_field_stress$top_sig), 
                              lab1="Control", lab2="Stress", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

Positive logFC which means **HIGHER** expression in tolerant than in sensitive cultivars
```{r venn diagram field-control-stress - tolerant vs. sensitive pos}
venn.plot <- func_venn_diagram_2(rownames(res_field_control$top_sig_up), 
                              rownames(res_field_stress$top_sig_up), 
                              lab1="Control", lab2="Stress", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

Negative logFC which means **LOWER** expression in tolerant than in sensitive cultivars
```{r venn diagram field-control-stress - tolerant vs. sensitive neg}
venn.plot <- func_venn_diagram_2(rownames(res_field_control$top_sig_down), 
                              rownames(res_field_stress$top_sig_down), 
                              lab1="Control", lab2="Stress", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

**Venn diagram of differentially expressed genes for sensitive vs. tolerant comparing samples from field and greenhouse**
**CONTROL SAMPLES**
Positive logFC which means **HIGHER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse-field-control - tolerant vs. sensitive pos}
venn.plot <- func_venn_diagram_2(rownames(res_field_control$top_sig_up), 
                              rownames(res_greenhouse_control$top_sig_up), 
                              lab1="Field", lab2="Greenhouse", filepath=NULL)
plot.new()
grid.draw(venn.plot)

head(intersect(rownames(res_field_control$top_sig_up), rownames(res_greenhouse_control$top_sig_up)))

```

**CONTROL SAMPLES**
Negative logFC which means **LOWER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse-field-control - tolerant vs. sensitive neg}
venn.plot <- func_venn_diagram_2(rownames(res_field_control$top_sig_down), 
                              rownames(res_greenhouse_control$top_sig_down), 
                              lab1="Field", lab2="Greenhouse", filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

**DROUGHT STRESS SAMPLES**
Positive logFC which means **HIGHER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse-field-stress - tolerant vs. sensitive pos}
venn.plot <- func_venn_diagram_2(rownames(res_field_stress$top_sig_up), 
                              rownames(res_greenhouse_stress$top_sig_up), 
                              lab1="Field", lab2="Greenhouse", filepath=NULL,
                              rot = 180)
# rotation.degree needs to be changed because the first area is smaller than second area!
plot.new()
grid.draw(venn.plot)
```

**DROUGHT STRESS SAMPLES**
Negative logFC which means **LOWER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse-field-stress - tolerant vs. sensitive neg}
venn.plot <- func_venn_diagram_2(rownames(res_field_stress$top_sig_down), 
                              rownames(res_greenhouse_stress$top_sig_down), 
                              lab1="Field", lab2="Greenhouse", filepath=NULL,
                              rot = 180) 
# rotation.degree needs to be changed because the first area is smaller than second area!
plot.new()
grid.draw(venn.plot)
```

**Compare 4 different DGE**
Positive logFC which means **HIGHER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse/field control/stress - tolerant vs. sensitive pos}
venn.plot <- func_venn_diagram_4(rownames(res_field_control$top_sig_up), 
                              rownames(res_field_stress$top_sig_up),
                              rownames(res_greenhouse_control$top_sig_up),
                              rownames(res_greenhouse_stress$top_sig_up),
                              lab1 = "Field \n Control", 
                              lab2 = "Field \n Stress",
                              lab3 = "Greenhouse \n Control", 
                              lab4 = "Greenhouse \n Stress", 
                              filepath=NULL)
plot.new()
grid.draw(venn.plot)
```

Negative logFC which means **LOWER** expression in tolerant than in sensitive cultivars
```{r venn diagram greenhouse/field control/stress - tolerant vs. sensitive neg}
venn.plot <- func_venn_diagram_4(rownames(res_field_control$top_sig_down), 
                              rownames(res_field_stress$top_sig_down),
                              rownames(res_greenhouse_control$top_sig_down),
                              rownames(res_greenhouse_stress$top_sig_down),
                              lab1 = "Field \n Control", 
                              lab2 = "Field \n Stress",
                              lab3 = "Greenhouse \n Control", 
                              lab4 = "Greenhouse \n Stress", 
                              filepath=NULL)
plot.new()
grid.draw(venn.plot)
```


### Save tables for PageMan (tolerant vs. sensitive)
columns: 
* pgsc_dmp
* logFC
* dir
+ 0: no significant change
+ 1: significantly higher
+ -1: significantly lower

```{r save tables for PageMan tolerant vs. sensitive}
res_greenhouse_control_pageman <- func_pageman_res(res_greenhouse_control)
res_greenhouse_stress_pageman <- func_pageman_res(res_greenhouse_stress)
res_field_control_pageman <- func_pageman_res(res_field_control)
res_field_stress_pageman <- func_pageman_res(res_field_stress)

head(res_field_stress_pageman)

write.table(res_greenhouse_control_pageman, "output/pageman_greenhouse_control_all.txt", 
            sep="\t", row.names=F, quote=F)
write.table(res_greenhouse_stress_pageman, "output/pageman_greenhouse_stress_all.txt", 
            sep="\t", row.names=F, quote=F)
write.table(res_field_control_pageman, "output/pageman_field_control_all.txt", 
            sep="\t", row.names=F, quote=F)
write.table(res_field_stress_pageman, "output/pageman_field_stress_all.txt", 
            sep="\t", row.names=F, quote=F)
```

#### Save tables for PageMan (tolerant vs. sensitive) WITH DMG
```{r save tables for PageMan (tolerant vs. sensitive) WITH DMG}
res_greenhouse_control_pageman_dmg <- func_pageman_res_dmg(res_greenhouse_control)
res_greenhouse_stress_pageman_dmg <- func_pageman_res_dmg(res_greenhouse_stress)
res_field_control_pageman_dmg <- func_pageman_res_dmg(res_field_control)
res_field_stress_pageman_dmg <- func_pageman_res_dmg(res_field_stress)
```


#### Merge MapMan to PageMan: tolerant vs. sensitive
```{r merge MapMan to PageMan tolerant vs. sensitive}
# merge to mapman bins
# dir ==  1 --> tol > sens --> T
# dir == -1 --> sens > tol --> S
res_field_control_pageman_mapman_T <- merge(res_field_control_pageman_dmg[which(res_field_control_pageman_dmg$dir == 1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
res_field_control_pageman_mapman_S <- merge(res_field_control_pageman_dmg[which(res_field_control_pageman_dmg$dir == -1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
res_field_stress_pageman_mapman_T <- merge(res_field_stress_pageman_dmg[which(res_field_stress_pageman_dmg$dir == 1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
res_field_stress_pageman_mapman_S <- merge(res_field_stress_pageman_dmg[which(res_field_stress_pageman_dmg$dir == -1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
# greenhouse
res_greenhouse_control_pageman_mapman_T <- merge(res_greenhouse_control_pageman_dmg[which(res_greenhouse_control_pageman_dmg$dir == 1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
res_greenhouse_control_pageman_mapman_S <- merge(res_greenhouse_control_pageman_dmg[which(res_greenhouse_control_pageman_dmg$dir == -1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
res_greenhouse_stress_pageman_mapman_T <- merge(res_greenhouse_stress_pageman_dmg[which(res_greenhouse_stress_pageman_dmg$dir == 1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")
res_greenhouse_stress_pageman_mapman_S <- merge(res_greenhouse_stress_pageman_dmg[which(res_greenhouse_stress_pageman_dmg$dir == -1),], 
                                            assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")

# sort
res_field_control_pageman_mapman_T <- func_order_by_MapMan_BinCode( droplevels(res_field_control_pageman_mapman_T), "BINCODE")
res_field_control_pageman_mapman_S <- func_order_by_MapMan_BinCode( droplevels(res_field_control_pageman_mapman_S), "BINCODE")
res_field_stress_pageman_mapman_T <- func_order_by_MapMan_BinCode( droplevels(res_field_stress_pageman_mapman_T), "BINCODE")
res_field_stress_pageman_mapman_S <- func_order_by_MapMan_BinCode( droplevels(res_field_stress_pageman_mapman_S), "BINCODE")

res_greenhouse_control_pageman_mapman_T <- func_order_by_MapMan_BinCode( droplevels(res_greenhouse_control_pageman_mapman_T), "BINCODE")
res_greenhouse_control_pageman_mapman_S <- func_order_by_MapMan_BinCode( droplevels(res_greenhouse_control_pageman_mapman_S), "BINCODE")
res_greenhouse_stress_pageman_mapman_T <- func_order_by_MapMan_BinCode( droplevels(res_greenhouse_stress_pageman_mapman_T), "BINCODE")
res_greenhouse_stress_pageman_mapman_S <- func_order_by_MapMan_BinCode( droplevels(res_greenhouse_stress_pageman_mapman_S), "BINCODE")

# save tables
write.table(res_field_control_pageman_mapman_T, "output/res_field_control_pageman_mapman_T.txt", sep="\t", row.names=F, quote=F)
write.table(res_field_control_pageman_mapman_S, "output/res_field_control_pageman_mapman_S.txt", sep="\t", row.names=F, quote=F)
write.table(res_field_stress_pageman_mapman_T, "output/res_field_stress_pageman_mapman_T.txt", sep="\t", row.names=F, quote=F)
write.table(res_field_stress_pageman_mapman_S, "output/res_field_stress_pageman_mapman_S.txt", sep="\t", row.names=F, quote=F)
write.table(res_greenhouse_control_pageman_mapman_T, "output/res_greenhouse_control_pageman_mapman_T.txt", sep="\t", row.names=F, quote=F)
write.table(res_greenhouse_control_pageman_mapman_S, "output/res_greenhouse_control_pageman_mapman_S.txt", sep="\t", row.names=F, quote=F)
write.table(res_greenhouse_stress_pageman_mapman_T, "output/res_greenhouse_stress_pageman_mapman_T.txt", sep="\t", row.names=F, quote=F)
write.table(res_greenhouse_stress_pageman_mapman_S, "output/res_greenhouse_stress_pageman_mapman_S.txt", sep="\t", row.names=F, quote=F)

# res_field_control_pageman_mapman_T <- res_field_control_pageman_mapman_T[order(res_field_control_pageman_mapman_T$BINCODE),]
# res_field_control_pageman_mapman_S <- res_field_control_pageman_mapman_S[order(res_field_control_pageman_mapman_S$BINCODE),]
```


#### PageMan tolerant vs. sensitive --> cytochrome P450
```{r PageMan tolerant vs. sensitive --> cytochrome P450}
# number of significant genes in bin 26.10*
# field
length(unique(subset(res_field_control_pageman_mapman_T$logFC, 
              grepl("26.10*",res_field_control_pageman_mapman_T$BINCODE)))) # control T>S
length(unique(subset(res_field_control_pageman_mapman_S$logFC, 
              grepl("26.10*",res_field_control_pageman_mapman_S$BINCODE)))) # control S>T
length(unique(subset(res_field_stress_pageman_mapman_T$logFC, 
              grepl("26.10*",res_field_stress_pageman_mapman_T$BINCODE)))) # stress T>S
length(unique(subset(res_field_stress_pageman_mapman_S$logFC, 
              grepl("26.10*",res_field_stress_pageman_mapman_S$BINCODE)))) # stress S>T
# greenhouse
length(unique(subset(res_greenhouse_control_pageman_mapman_T$logFC, 
              grepl("26.10*",res_greenhouse_control_pageman_mapman_T$BINCODE)))) # control T>S
length(unique(subset(res_greenhouse_control_pageman_mapman_S$logFC, 
              grepl("26.10*",res_greenhouse_control_pageman_mapman_S$BINCODE)))) # control S>T
length(unique(subset(res_greenhouse_stress_pageman_mapman_T$logFC, 
              grepl("26.10*",res_greenhouse_stress_pageman_mapman_T$BINCODE)))) # stress T>S
length(unique(subset(res_greenhouse_stress_pageman_mapman_S$logFC, 
              grepl("26.10*",res_greenhouse_stress_pageman_mapman_S$BINCODE)))) # stress S>T

#intersect(res_field_control_pageman_mapman_T$pgsc_dmp[which(grepl("26.10*",res_field_control_pageman_mapman_T$BINCODE))],
#          res_field_control_pageman_mapman_S$pgsc_dmp[which(grepl("26.10*",res_field_control_pageman_mapman_S$BINCODE))])
```


#### PageMan tolerant vs. sensitive --> receptor kinases
```{r PageMan tolerant vs. sensitive --> receptor kinases}
# number of significant genes in bin 30.2*
# field
length(unique(subset(res_field_control_pageman_mapman_T$logFC, 
              grepl("30.2*",res_field_control_pageman_mapman_T$BINCODE))))
length(unique(subset(res_field_control_pageman_mapman_S$logFC, 
              grepl("30.2*",res_field_control_pageman_mapman_S$BINCODE))))
length(unique(subset(res_field_stress_pageman_mapman_T$logFC, 
              grepl("30.2*",res_field_stress_pageman_mapman_T$BINCODE))))
length(unique(subset(res_field_stress_pageman_mapman_S$logFC, 
              grepl("30.2*",res_field_stress_pageman_mapman_S$BINCODE))))
# greenhouse
length(unique(subset(res_greenhouse_control_pageman_mapman_T$logFC, 
              grepl("30.2*",res_greenhouse_control_pageman_mapman_T$BINCODE))))
length(unique(subset(res_greenhouse_control_pageman_mapman_S$logFC, 
              grepl("30.2*",res_greenhouse_control_pageman_mapman_S$BINCODE))))
length(unique(subset(res_greenhouse_stress_pageman_mapman_T$logFC, 
              grepl("30.2*",res_greenhouse_stress_pageman_mapman_T$BINCODE))))
length(unique(subset(res_greenhouse_stress_pageman_mapman_S$logFC, 
              grepl("30.2*",res_greenhouse_stress_pageman_mapman_S$BINCODE))))
```


#### FDR and logFC --> only receptor kinases (30.2)
```{r FDR and logFC --> only receptor kinases (30.2)}
length(which(grepl("30.2*",merge_mapman_pgsc$BINCODE)))
mapman_kinases <- merge_mapman_pgsc[which(grepl("30.2*",merge_mapman_pgsc$BINCODE)),]
                               
res_field_control_kinases <- func_merge_res_stress_bin(res_field_control$top_all, mapman_kinases)
res_field_stress_kinases <- func_merge_res_stress_bin(res_field_stress$top_all, mapman_kinases)
res_greenhouse_control_kinases <- func_merge_res_stress_bin(res_greenhouse_control$top_all, mapman_kinases)
res_greenhouse_stress_kinases <- func_merge_res_stress_bin(res_greenhouse_stress$top_all, mapman_kinases)

fdr_kinases <- cbind(field_control = res_field_control_kinases$FDR,
                     field_stress = res_field_stress_kinases$FDR,
                     greenhouse_control = res_greenhouse_control_kinases$FDR,
                     greenhouse_stress = res_greenhouse_stress_kinases$FDR)
fdr_kinases_min <- apply(fdr_kinases, 1, min)
length(which(fdr_kinases_min<0.001))

logfc_kinases <- cbind(field_control = res_field_control_kinases$logFC,
                     field_stress = res_field_stress_kinases$logFC,
                     greenhouse_control = res_greenhouse_control_kinases$logFC,
                     greenhouse_stress = res_greenhouse_stress_kinases$logFC)

rownames(logfc_kinases) <- res_field_stress_kinases$pgsc_dmg
logfc_kinases_max <- abs(apply(logfc_kinases, 1, max))
logfc_kinases_part <- logfc_kinases[which(fdr_kinases_min < 0.001),]
#logfc_kinases_part <- logfc_kinases[which(logfc_kinases_max > 1),]
#hist(logfc_kinases_part)
logfc_kinases_part <- unique(merge(logfc_kinases_part, assoc_pgsc[,c(1,5)], by.x="row.names", by.y="pgsc_dmg"))
dim(logfc_kinases_part)
```


#### heatmap only receptor kinases bin
```{r heatmap only receptor kinases bin, eval=FALSE}
mat_data <- as.matrix(logfc_kinases_part[,c(2:5)])
rownames(mat_data) <- logfc_kinases_part[,6]
distance = dist(mat_data, method = "euclidian")
cluster = hclust(distance, method = "average")
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 299)
col_breaks = c(seq(-4,-1,length=100), # for blue
               seq(-1,1,length=100), # for white
               seq(1,4,length=100)) # for red

pdf("figures/heatmap_log2FC_kinases.pdf", width=6, height=15)

lmat=rbind(4:3, 2:1)
lhei=c(0.4, 5)
lwid=c(1, 4)

layout(mat = lmat, widths = lwid, heights = lhei)

heatmap.2(mat_data,
  #cellnote = signif2,  # same data set for cell labels
  #notecex=1.5, # numeric scaling factor for cellnote items
  colsep=1:ncol(mat_data), 
  rowsep=1:nrow(mat_data),
  sepwidth=c(0.01,0.05),
  sepcolor="black",
  main = "", # heat map title
  #notecol="black",      # change font color of cell labels to black
  density.info="none",  # turns off density plot inside color legend
  trace="none",         # turns off trace lines inside the heat map
  margins =c(8,18),     # widens margins around plot
  keysize=1.5,
  cexRow=1.2,
  cexCol=1.5,
  adjCol=c(0,1),
  srtCol = 330,
  lhei=c(0.4, 5),
  col=my_palette,       # use on color palette defined earlier 
  breaks=col_breaks,    # enable color transition at specified limits
  #RowSideColors=mycolor,
  #ColSideColors=column_color,
  Rowv=as.dendrogram(cluster),
  dendrogram="row",     # only draw a row dendrogram
  Colv="NA")            # turn off column clustering```

dev.off()
```


## Control vs. Drought Stress
### GREENHOUSE + TOLERANT: Control vs. Drought Stress
Execute edgeR procedure to identify differentially expresses genes (DGE) comparing control and drought stress conditions for field samples
```{r edgeR greenhouse tolerant - control vs. stress, echo=TRUE}
group_idx <- which(samplelist_modified[, "cultivation_tolerance"]=="greenhouse.tolerant")
group_factor <- as.factor(subset(samplelist_modified, cultivation_tolerance == "greenhouse.tolerant")$condition)

res_greenhouse_tolerant <- func_edgeR_dge(counts_keep, group_idx, group_1 = group_factor)
res_greenhouse_tolerant$sums
head(res_greenhouse_tolerant$top_sig)
```

### GREENHOUSE + SENSITIVE: Control vs. Drought Stress
Execute edgeR procedure to identify differentially expresses genes (DGE) comparing control and drought stress conditions for field samples
```{r edgeR greenhouse sensitive - control vs. stress, echo=TRUE}
group_idx <- which(samplelist_modified[, "cultivation_tolerance"]=="greenhouse.sensitive")
group_factor <- as.factor(subset(samplelist_modified, cultivation_tolerance == "greenhouse.sensitive")$condition)

res_greenhouse_sensitive <- func_edgeR_dge(counts_keep, group_idx, group_1 = group_factor)
res_greenhouse_sensitive$sums
head(res_greenhouse_sensitive$top_sig)
```

### FIELD + TOLERANT: Control vs. Drought Stress
Execute edgeR procedure to identify differentially expresses genes (DGE) comparing control and drought stress conditions for field samples
```{r edgeR field tolerant - control vs. stress, echo=TRUE}
group_idx <- which(samplelist_modified[, "cultivation_tolerance"]=="field.tolerant")
group_factor <- as.factor(subset(samplelist_modified, cultivation_tolerance == "field.tolerant")$condition)

res_field_tolerant <- func_edgeR_dge(counts_keep, group_idx, group_1 = group_factor)
res_field_tolerant$sums
head(res_field_tolerant$top_sig)
```

### FIELD + SENSITIVE: Control vs. Drought Stress
Execute edgeR procedure to identify differentially expresses genes (DGE) comparing control and drought stress conditions for field samples
```{r edgeR field sensitive - control vs. stress, echo=TRUE}
group_idx <- which(samplelist_modified[, "cultivation_tolerance"]=="field.sensitive")
group_factor <- as.factor(subset(samplelist_modified, cultivation_tolerance == "field.sensitive")$condition)

res_field_sensitive <- func_edgeR_dge(counts_keep, group_idx, group_1 = group_factor)
res_field_sensitive$sums
head(res_field_sensitive$top_sig)
```

#### Intersection greenhouse-field-TOLERANT control vs. stress
```{r intersection greenhouse-field-TOLERANT control vs. stress}
# up-regulated genes
intersect_tolerant_up <- 
  func_intersecting_dge(set1 = res_field_tolerant$top_sig_up, 
                        set2 = res_greenhouse_tolerant$top_sig_up, 
                        lab1 = "field", lab2 = "gh")

# first 25 intersecting DGE (columns: row.names, logFC of set1, logFC of set2, logFC mean, logFC sum, annotations)
pander(head(  intersect_tolerant_up[,c(1,2,6,10:13)], n=25))

write.table(intersect_tolerant_up, "output/intersect_tolerant_up.txt", sep="\t", row.names=F, quote=F)

# venn diagram
venn.plot <- func_venn_diagram_2(rownames(res_field_tolerant$top_sig_up), 
                              rownames(res_greenhouse_tolerant$top_sig_up), 
                              lab1="Field", lab2="Greenhouse", filepath=NULL,
                              rot = 180)
plot.new()
grid.draw(venn.plot)

pdf("figures/venn_tolerant_greenhouse_field_up.pdf")
grid.draw(venn.plot)
dev.off()

#############################################

# down-regulated genes
intersect_tolerant_down <- 
  func_intersecting_dge(set1 = res_field_tolerant$top_sig_down, 
                        set2 = res_greenhouse_tolerant$top_sig_down, 
                        lab1 = "field", lab2 = "gh")

# no intersection!

# pdf("figures/venn_tolerant_greenhouse_field_down.pdf")
# grid.draw(venn.plot)
# dev.off()
```

#### Intersection greenhouse-field-SENSITIVE control vs. stress
```{r intersection greenhouse-field-SENSITIVE control vs. stress}
# up-regulated genes
intersect_sensitive_up <- 
  func_intersecting_dge(set1 = res_field_sensitive$top_sig_up, 
                        set2 = res_greenhouse_sensitive$top_sig_up, 
                        lab1 = "field", lab2 = "gh")

# first 25 intersecting DGE (columns: row.names, logFC of set1, logFC of set2, logFC mean, logFC sum, annotations)
pander(head(  intersect_sensitive_up[,c(1,2,6,10:13)], n=25))

write.table(intersect_sensitive_up, "output/intersect_sensitive_up.txt", sep="\t", row.names=F, quote=F)

# venn diagram
venn.plot <- func_venn_diagram_2(rownames(res_field_sensitive$top_sig_up), 
                              rownames(res_greenhouse_sensitive$top_sig_up),
                              lab1 = "Field", lab2 = "Greenhouse", filepath = NULL,
                              rot = 180)
plot.new()
grid.draw(venn.plot)

pdf("figures/venn_sensitive_greenhouse_field_up.pdf")
grid.draw(venn.plot)
dev.off()


#############################################

# down-regulated genes
intersect_sensitive_down <- 
  func_intersecting_dge(set1 = res_field_sensitive$top_sig_down, 
                        set2 = res_greenhouse_sensitive$top_sig_down, 
                        lab1 = "field", lab2 = "gh")

# first 25 intersecting DGE (columns: row.names, logFC of set1, logFC of set2, logFC mean, logFC sum, annotations)
pander(head(  intersect_sensitive_down[,c(1,2,6,10:13)], n=25))

write.table(intersect_sensitive_down, "output/intersect_sensitive_down.txt", sep="\t", row.names=F, quote=F)

# venn diagram
venn.plot <- func_venn_diagram_2(rownames(res_field_sensitive$top_sig_down), 
                              rownames(res_greenhouse_sensitive$top_sig_down), 
                              lab1 = "Field", lab2 = "Greenhouse", filepath = NULL,
                              rot = 180)
plot.new()
grid.draw(venn.plot)

pdf("figures/venn_sensitive_greenhouse_field_down.pdf")
grid.draw(venn.plot)
dev.off()
```

#### Intersection sensitive-tolerant-GREENHOUSE control vs. stress
```{r intersection sensitive-tolerant-GREENHOUSE control vs. stress}
# up-regulated genes
intersect_greenhouse_up <- 
  func_intersecting_dge(set1 = res_greenhouse_tolerant$top_sig_up, 
                        set2 = res_greenhouse_sensitive$top_sig_up, 
                        lab1 = "tolerant", lab2 = "sensitive")

# first 25 intersecting DGE (columns: row.names, logFC of set1, logFC of set2, logFC mean, logFC sum, annotations)
pander(head(  intersect_greenhouse_up[,c(1,2,6,10:13)], n=25))

write.table(intersect_greenhouse_up, "output/intersect_greenhouse_up.txt", sep="\t", row.names=F, quote=F)

# venn diagram
venn.plot <- func_venn_diagram_2(rownames(res_greenhouse_sensitive$top_sig_up), 
                                 rownames(res_greenhouse_tolerant$top_sig_up),
                                 lab1 = "Sensitive", lab2 = "Tolerant", filepath = NULL)
plot.new()
grid.draw(venn.plot)

pdf("figures/venn_greenhouse_sensitive_tolerant_up.pdf")
grid.draw(venn.plot)
dev.off()

#############################################

# down-regulated genes
intersect_greenhouse_down <- 
  func_intersecting_dge(set1 = res_greenhouse_tolerant$top_sig_down, 
                        set2 = res_greenhouse_sensitive$top_sig_down, 
                        lab1 = "tolerant", lab2 = "sensitive")

# first 25 intersecting DGE (columns: row.names, logFC of set1, logFC of set2, logFC mean, logFC sum, annotations)
pander(head(  intersect_greenhouse_down[,c(1,2,6,10:13)], n=25))

write.table(intersect_greenhouse_down, "output/intersect_greenhouse_down.txt", sep="\t", row.names=F, quote=F)

# venn diagram
venn.plot <- func_venn_diagram_2(rownames(res_greenhouse_sensitive$top_sig_down), 
                              rownames(res_greenhouse_tolerant$top_sig_down),
                              lab1="Sensitive", lab2="Tolerant", filepath=NULL)
plot.new()
grid.draw(venn.plot)

pdf("figures/venn_greenhouse_sensitive_tolerant_down.pdf")
grid.draw(venn.plot)
dev.off()
```

#### Intersection sensitive-tolerant-FIELD control vs. stress
```{r intersection sensitive-tolerant-FIELD control vs. stress}
# up-regulated genes
intersect_field_up <- 
  func_intersecting_dge(set1 = res_field_tolerant$top_sig_up, 
                        set2 = res_field_sensitive$top_sig_up, 
                        lab1 = "tolerant", lab2 = "sensitive")

# first 25 intersecting DGE (columns: row.names, logFC of set1, logFC of set2, logFC mean, logFC sum, annotations)
pander(head(  intersect_field_up[,c(1,2,6,10:13)], n=25))

write.table(intersect_field_up, "output/intersect_field_up.txt", sep="\t", row.names=F, quote=F)

# venn diagram
venn.plot <- func_venn_diagram_2(rownames(res_field_sensitive$top_sig_up), 
                              rownames(res_field_tolerant$top_sig_up), 
                              lab1="Sensitive", lab2="Tolerant", filepath=NULL)
plot.new()
grid.draw(venn.plot)

pdf("figures/venn_field_sensitive_tolerant_up.pdf")
grid.draw(venn.plot)
dev.off()

#############################################

# down-regulated genes
intersect_field_down <- 
  func_intersecting_dge(set1 = res_field_tolerant$top_sig_down, 
                        set2 = res_field_sensitive$top_sig_down, 
                        lab1 = "tolerant", lab2 = "sensitive")

# first 25 intersecting DGE (columns: row.names, logFC of set1, logFC of set2, logFC mean, logFC sum, annotations)
pander(head(  intersect_field_down[,c(1,2,6,10:13)], n=25))

write.table(intersect_field_down, "output/intersect_field_down.txt", sep="\t", row.names=F, quote=F)

# venn diagram
venn.plot <- func_venn_diagram_2(rownames(res_field_sensitive$top_sig_down), 
                              rownames(res_field_tolerant$top_sig_down),
                              lab1="Sensitive", lab2="Tolerant", filepath=NULL)
plot.new()
grid.draw(venn.plot)

pdf("figures/venn_field_sensitive_tolerant_down.pdf")
grid.draw(venn.plot)
dev.off()
```


#### Intersection sensitive-tolerant-field-greenhouse control vs. stress UP
**Compare 4 different DGE**
Positive logFC which means **HIGHER** expression under stress than under control conditions
```{r venn diagram greenhouse/field tolerant/sensitive control vs. stress UP}
venn.plot <- func_venn_diagram_4(rownames(res_field_sensitive$top_sig_up), 
                              rownames(res_field_tolerant$top_sig_up),
                              rownames(res_greenhouse_sensitive$top_sig_up),
                              rownames(res_greenhouse_tolerant$top_sig_up),
                              lab1 = "Field \n Sensitive", 
                              lab2 = "Field \n Tolerant",
                              lab3 = "Greenhouse \n Sensitive", 
                              lab4 = "Greenhouse \n Tolerant", 
                              filepath=NULL)
plot.new()
grid.draw(venn.plot)

pdf("figures/venn_field_greenhouse_sensitive_tolerant_up.pdf")
grid.draw(venn.plot)
dev.off()
```

Negative logFC which means **LOWER** expression under stress than under control conditions
```{r venn diagram greenhouse/field tolerant/sensitive control vs. stress DOWN}
venn.plot <- func_venn_diagram_4(rownames(res_field_sensitive$top_sig_down), 
                              rownames(res_field_tolerant$top_sig_down),
                              rownames(res_greenhouse_sensitive$top_sig_down),
                              rownames(res_greenhouse_tolerant$top_sig_down),
                              lab1 = "Field \n Sensitive", 
                              lab2 = "Field \n Tolerant",
                              lab3 = "Greenhouse \n Sensitive", 
                              lab4 = "Greenhouse \n Tolerant", 
                              filepath=NULL)
plot.new()
grid.draw(venn.plot)

pdf("figures/venn_field_greenhouse_sensitive_tolerant_down.pdf")
grid.draw(venn.plot)
dev.off()
```


### Export for MapMan and PageMan (control vs. stress)
#### save tables for MapMan control vs. stress
```{r save tables for MapMan (control vs. stress)}
# only entries with FDR < 0.05
res_greenhouse_merge_sig <- subset(res_greenhouse$merge, res_greenhouse$merge$FDR < 0.05)
res_field_merge_sig <- subset(res_field$merge, res_field$merge$FDR < 0.05)

colnames(res_greenhouse_merge_sig)

# select columns: pgsc_dmp (8), logFC (2), pvalue (4), FDR (5)
write.table(res_greenhouse_merge_sig[,c(8,2,4,5)], "output/mapman_greenhouse_sig.txt", sep="\t", row.names=F, quote=F)
write.table(res_greenhouse$merge[,c(8,2,4,5)], "output/mapman_greenhouse_all.txt", sep="\t", row.names=F, quote=F)
write.table(res_field_merge_sig[,c(8,2,4,5)], "output/mapman_field_sig.txt", sep="\t", row.names=F, quote=F)
write.table(res_field$merge[,c(8,2,4,5)], "output/mapman_field_all.txt", sep="\t", row.names=F, quote=F)

# combine intersection of significantly up-/down-regulated genes for MapMan visualisation
# columns for PGSC_DMP if (12) and logFC (mean of greenhouse and field, 10)
colnames(intersect_cultivation_condition_up_res_complete)
head(intersect_cultivation_condition_up_res_complete[,c(12,10)])

write.table( rbind( unique(intersect_cultivation_condition_up_res_complete[,c(12,10)]),
                    unique(intersect_cultivation_condition_down_res_complete[,c(12,10)])), 
            "output/mapman_intersect_cultivation_condition_sig.txt",
             sep="\t", row.names=F, quote=F)

# write.table(res_greenhouse$top_all, "output/res_greenhouse_top_all.txt", sep="\t")
# write.table(res_greenhouse_control$top_all, "output/res_greenhouse_control_top_all.txt", sep="\t")
# write.table(res_greenhouse_stress$top_all, "output/res_greenhouse_stress_top_all.txt", sep="\t")
# write.table(res_field$top_all, "output/res_field_top_all.txt", sep="\t")
# write.table(res_field_control$top_all, "output/res_field_control_top_all.txt", sep="\t")
# write.table(res_field_stress$top_all, "output/res_field_stress_top_all.txt", sep="\t")
# 
# write.table(res_greenhouse$merge[,c(8,2,4,5)], "output/res_greenhouse_merge.txt", sep="\t")
# write.table(res_field$merge[,c(8,2,4,5)], "output/res_field_merge.txt", sep="\t")
```


#### save tables for PageMan control vs. stress
```{r save tables for PageMan (control vs. stress)}
# results together for all cultivars
res_greenhouse_pageman <- func_pageman_res(res_greenhouse)
res_field_pageman <- func_pageman_res(res_field)
table(res_field_pageman$dir)

# results separately for tolerant and sensitive cultivars
res_field_tolerant_pageman <- func_pageman_res(res_field_tolerant)
res_field_sensitive_pageman <- func_pageman_res(res_field_sensitive)
res_greenhouse_tolerant_pageman <- func_pageman_res(res_greenhouse_tolerant)
res_greenhouse_sensitive_pageman <- func_pageman_res(res_greenhouse_sensitive)

write.table(res_greenhouse_pageman, "output/pageman_greenhouse_all.txt", sep="\t", row.names=F, quote=F)
write.table(res_field_pageman, "output/pageman_field_all.txt", sep="\t", row.names=F, quote=F)

write.table(res_greenhouse_tolerant_pageman, 
            "output/pageman_greenhouse_tolerant_all.txt", 
            sep="\t", row.names=F, quote=F)
write.table(res_greenhouse_sensitive_pageman, 
            "output/pageman_greenhouse_sensitive_all.txt", 
            sep="\t", row.names=F, quote=F)
write.table(res_field_tolerant_pageman, 
            "output/pageman_field_tolerant_all.txt", 
            sep="\t", row.names=F, quote=F)
write.table(res_field_sensitive_pageman, 
            "output/pageman_field_sensitive_all.txt", 
            sep="\t", row.names=F, quote=F)

# pageman_greenhouse_tolerant_sensitive <- cbind(res_greenhouse_tolerant_pageman, 
#                                                res_greenhouse_sensitive_pageman[,-1])
# colnames(pageman_greenhouse_tolerant_sensitive)[2:5] <- c("logFC_tol", "dir_tol", "logFC_sens", "dir_sens")
# head(pageman_greenhouse_tolerant_sensitive)
# 
# write.table(pageman_greenhouse_tolerant_sensitive, 
#             "output/pageman_greenhouse_tolerant_sensitive_all.txt", 
#             sep="\t", row.names=F, quote=F)



# use only significantly changed genes (DGEs): dir != 0
res_greenhouse_tolerant_pageman_sig <- subset(res_greenhouse_tolerant_pageman, res_greenhouse_tolerant_pageman$dir != 0)
res_greenhouse_sensitive_pageman_sig <- subset(res_greenhouse_sensitive_pageman, res_greenhouse_sensitive_pageman$dir != 0)

hist(res_greenhouse_tolerant_pageman_sig$logFC, col="grey", breaks=20)
hist(res_greenhouse_sensitive_pageman_sig$logFC, col="grey", breaks=20)

length(which(abs(res_greenhouse_sensitive_pageman_sig$logFC)<0.5))

write.table(res_greenhouse_tolerant_pageman_sig, 
            "output/pageman_greenhouse_tolerant_sig.txt", 
            sep="\t", row.names=F, quote=F)

write.table(res_greenhouse_sensitive_pageman_sig, 
            "output/pageman_greenhouse_sensitive_sig.txt", 
            sep="\t", row.names=F, quote=F)
```

#### Save tables for PageMan control vs. stress WITH DMG
```{r save tables for PageMan control vs. stress WITH DMG}
# results separately for tolerant and sensitive cultivars
res_field_tolerant_pageman_dmg <- func_pageman_res_dmg(res_field_tolerant)
res_field_sensitive_pageman_dmg <- func_pageman_res_dmg(res_field_sensitive)
res_greenhouse_tolerant_pageman_dmg <- func_pageman_res_dmg(res_greenhouse_tolerant)
res_greenhouse_sensitive_pageman_dmg <- func_pageman_res_dmg(res_greenhouse_sensitive)

head(res_greenhouse_sensitive_pageman_dmg)
```


#### Merge significant Pageman res to MapMan bins field
```{r merge significant Pageman res to MapMan bins field (control vs. stress)}
# only significant DGE: dir must be -1 or +1 (not 0)!
# tolerant
res_field_tolerant_pageman_mapman <- merge(
  res_field_tolerant_pageman_dmg[which(res_field_tolerant_pageman_dmg$dir%in%c(-1,1)),], 
  assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")

# sensitive
res_field_sensitive_pageman_mapman <- merge(
  res_field_sensitive_pageman_dmg[which(res_field_sensitive_pageman_dmg$dir%in%c(-1,1)),], 
  assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")

# sort
res_field_tolerant_pageman_mapman <- func_order_by_MapMan_BinCode( droplevels(res_field_tolerant_pageman_mapman), "BINCODE")
res_field_sensitive_pageman_mapman <- func_order_by_MapMan_BinCode( droplevels(res_field_sensitive_pageman_mapman), "BINCODE")

#res_field_sensitive_pageman_mapman <- merge(res_field_sensitive_pageman, assoc_mapman, 
#                                           by.x = "pgsc_dmp", by.y = "IDENTIFIER")
#write.table(res_field_sensitive_pageman_mapman, "../output/res_field_sensitive_pageman_mapman.txt", 
#            sep="\t", row.names=F, quote=F)

write.table(res_field_tolerant_pageman_mapman,
            "output/res_field_tolerant_pageman_mapman.txt",
            sep="\t", row.names=F, quote=F)

write.table(res_field_sensitive_pageman_mapman,
            "output/res_field_sensitive_pageman_mapman.txt",
            sep="\t", row.names=F, quote=F)

table(droplevels(res_field_sensitive_pageman_mapman$BIN))
table(droplevels(res_field_tolerant_pageman_mapman$BIN))

# only bin 20.2.1: "stress.abiotic.heat"
subset(res_field_sensitive_pageman_mapman[,1:5],
              res_field_sensitive_pageman_mapman$BIN == "20.2.1")

# sensitive
length(unique(subset(res_field_sensitive_pageman_mapman$logFC, 
              res_field_sensitive_pageman_mapman$BIN == "20.2.1")))

# tolerant
length(unique(subset(res_field_tolerant_pageman_mapman$logFC, 
              res_field_tolerant_pageman_mapman$BIN == "20.2.1")))

# only bin 29.5.7: "protein.degradation.metalloprotease"
subset(res_field_tolerant_pageman_mapman[,1:5],  
              res_field_tolerant_pageman_mapman$BIN == "29.5.7")

# order table by BINCODE
res_field_sensitive_pageman_mapman_ordered <- res_field_sensitive_pageman_mapman[order(res_field_sensitive_pageman_mapman$BINCODE),c(1:4)]
head(res_field_sensitive_pageman_mapman_ordered)
```


#### merge significant Pageman res to MapMan bins greenhouse
```{r merge significant Pageman res to MapMan bins greenhouse (control vs. stress)}
# only significant DGE: dir must be -1 or +1 (not 0)!
# tolerant
res_greenhouse_tolerant_pageman_mapman <- merge(
  res_greenhouse_tolerant_pageman_dmg[which(res_greenhouse_tolerant_pageman_dmg$dir%in%c(-1,1)),], 
  assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")

# sensitive
res_greenhouse_sensitive_pageman_mapman <- merge(
  res_greenhouse_sensitive_pageman_dmg[which(res_greenhouse_sensitive_pageman_dmg$dir%in%c(-1,1)),],
  assoc_mapman, by.x = "pgsc_dmp", by.y = "IDENTIFIER")

# sort
res_greenhouse_tolerant_pageman_mapman <- func_order_by_MapMan_BinCode( droplevels(res_greenhouse_tolerant_pageman_mapman), "BINCODE")
res_greenhouse_sensitive_pageman_mapman <- func_order_by_MapMan_BinCode( droplevels(res_greenhouse_sensitive_pageman_mapman), "BINCODE")


write.table(res_greenhouse_tolerant_pageman_mapman,
            "output/res_greenhouse_tolerant_pageman_mapman.txt",
            sep="\t", row.names=F, quote=F)

write.table(res_greenhouse_sensitive_pageman_mapman,
            "output/res_greenhouse_sensitive_pageman_mapman.txt",
            sep="\t", row.names=F, quote=F)


table(droplevels(res_greenhouse_sensitive_pageman_mapman$BIN))
table(droplevels(res_greenhouse_tolerant_pageman_mapman$BIN))

# only bin 20.2.1: "stress.abiotic.heat"
length(unique(subset(res_greenhouse_sensitive_pageman_mapman$logFC, 
              res_greenhouse_sensitive_pageman_mapman$BIN == "20.2.1")))

length(unique(subset(res_greenhouse_tolerant_pageman_mapman$logFC, 
              res_greenhouse_tolerant_pageman_mapman$BIN == "20.2.1")))
```

#### merge edgeR-results to mapman bins
```{r merge edgeR-results to mapman bins (control vs. stress)}
# use only DGE in results table
res_field_tolerant_mapman <- func_merge_sig_res_mapman(res_field_tolerant$top_sig)
res_field_sensitive_mapman <- func_merge_sig_res_mapman(res_field_sensitive$top_sig)
res_greenhouse_tolerant_mapman <- func_merge_sig_res_mapman(res_greenhouse_tolerant$top_sig)
res_greenhouse_sensitive_mapman <- func_merge_sig_res_mapman(res_greenhouse_sensitive$top_sig)

# use all genes in results table
res_greenhouse_tolerant_mapman_all_up <- func_merge_sig_res_mapman(res_greenhouse_tolerant$top_all_up)
res_greenhouse_tolerant_mapman_all_down <- func_merge_sig_res_mapman(res_greenhouse_tolerant$top_all_down)
res_greenhouse_tolerant$sums
```


### only abiotic stress bin 20.2 (control vs. stress)
```{r only abiotic stress bin 20.2 (control vs. stress)}
length(which(grepl("20.2.1*", merge_mapman_pgsc$BINCODE)))
mapman_abiotic_stress <- merge_mapman_pgsc[which(grepl("20.2.1*", merge_mapman_pgsc$BINCODE)),]
                               
res_field_tolerant_abiotic_stress <- func_merge_res_stress_bin(res_field_tolerant$top_all, mapman_abiotic_stress)
res_field_sensitive_abiotic_stress <- func_merge_res_stress_bin(res_field_sensitive$top_all, mapman_abiotic_stress)
res_greenhouse_tolerant_abiotic_stress <- func_merge_res_stress_bin(res_greenhouse_tolerant$top_all, mapman_abiotic_stress)
res_greenhouse_sensitive_abiotic_stress <- func_merge_res_stress_bin(res_greenhouse_sensitive$top_all, mapman_abiotic_stress)

fdr_abiotic_stress <- cbind(field_tol = res_field_tolerant_abiotic_stress$FDR,
                            field_sens = res_field_sensitive_abiotic_stress$FDR,
                            greenhouse_tol = res_greenhouse_tolerant_abiotic_stress$FDR,
                            greenhouse_sens = res_greenhouse_sensitive_abiotic_stress$FDR)
fdr_abiotic_stress_min <- apply(fdr_abiotic_stress, 1, min)
length(which(fdr_abiotic_stress_min<0.05))

logfc_abiotic_stress <- cbind(field_tol = res_field_tolerant_abiotic_stress$logFC,
                            field_sens = res_field_sensitive_abiotic_stress$logFC,
                            greenhouse_tol = res_greenhouse_tolerant_abiotic_stress$logFC,
                            greenhouse_sens = res_greenhouse_sensitive_abiotic_stress$logFC)

rownames(logfc_abiotic_stress) <- res_field_sensitive_abiotic_stress$pgsc_dmg
logfc_abiotic_stress_max <- abs(apply(logfc_abiotic_stress, 1, max))
#logfc_abiotic_stress_part <- logfc_abiotic_stress[which(fdr_abiotic_stress_min < 0.05),]
logfc_abiotic_stress_part <- logfc_abiotic_stress[which(logfc_abiotic_stress_max > 1),]
hist(logfc_abiotic_stress_part)
```


### heatmap only abiotic stress bin 20.2 (control vs. stress)
```{r heatmap only abiotic stress bin 20.2 (control vs. stress)}
# mat_data <- as.matrix(logfc_abiotic_stress_part)
# 
# distance = dist(mat_data, method = "euclidian")
# cluster = hclust(distance, method = "average")
# 
# my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 299)
# col_breaks = c(seq(-5,-1,length=100), # for blue
#                seq(-1,1,length=100), # for white
#                seq(1,5,length=100)) # for red
# 
# pdf("figures/heatmap_log2FC_abiotic_stress.pdf", width=6, height=15)
# 
# lmat=rbind(4:3, 2:1)
# lhei=c(0.4, 5)
# lwid=c(1, 4)
# 
# layout(mat = lmat, widths = lwid, heights = lhei)
# 
# heatmap.2(mat_data,
#   #cellnote = signif2,  # same data set for cell labels
#   #notecex=1.5, # numeric scaling factor for cellnote items
#   colsep=1:ncol(mat_data), 
#   rowsep=1:nrow(mat_data),
#   sepwidth=c(0.01,0.05),
#   sepcolor="black",
#   main = "", # heat map title
#   #notecol="black",      # change font color of cell labels to black
#   density.info="none",  # turns off density plot inside color legend
#   trace="none",         # turns off trace lines inside the heat map
#   margins =c(8,18),     # widens margins around plot
#   keysize=1.5,
#   cexRow=1.2,
#   cexCol=1.5,
#   adjCol=c(0,1),
#   srtCol = 330,
#   lhei=c(0.4, 5),
#   col=my_palette,       # use on color palette defined earlier 
#   breaks=col_breaks,    # enable color transition at specified limits
#   #RowSideColors=mycolor,
#   #ColSideColors=column_color,
#   Rowv=as.dendrogram(cluster),
#   dendrogram="row",     # only draw a row dendrogram
#   Colv="NA")            # turn off column clustering```
# 
# dev.off()
```


### only protein bin 29 (control vs. stress)
```{r only protein bin 29 (control vs. stress)}
length(which(grepl("protein*",merge_mapman_pgsc$NAME)))
mapman_protein <- merge_mapman_pgsc[which(grepl("protein*",merge_mapman_pgsc$NAME)),]
                               
res_field_tolerant_protein <- func_merge_res_stress_bin(res_field_tolerant$top_all, mapman_protein)
res_field_sensitive_protein <- func_merge_res_stress_bin(res_field_sensitive$top_all, mapman_protein)
res_greenhouse_tolerant_protein <- func_merge_res_stress_bin(res_greenhouse_tolerant$top_all, mapman_protein)
res_greenhouse_sensitive_protein <- func_merge_res_stress_bin(res_greenhouse_sensitive$top_all, mapman_protein)

fdr_protein <- cbind(field_tol = res_field_tolerant_protein$FDR,
                            field_sens = res_field_sensitive_protein$FDR,
                            greenhouse_tol = res_greenhouse_tolerant_protein$FDR,
                            greenhouse_sens = res_greenhouse_sensitive_protein$FDR)
fdr_protein_min <- apply(fdr_protein, 1, min)
length(which(fdr_protein_min<0.05))

logfc_protein <- cbind(field_tol = res_field_tolerant_protein$logFC,
                            field_sens = res_field_sensitive_protein$logFC,
                            greenhouse_tol = res_greenhouse_tolerant_protein$logFC,
                            greenhouse_sens = res_greenhouse_sensitive_protein$logFC)
rownames(logfc_protein) <- res_field_sensitive_protein$pgsc_dmg
logfc_protein_max <- abs(apply(logfc_protein, 1, max))
logfc_protein_part <- logfc_protein[which(fdr_protein_min<0.05),]
hist(logfc_protein_part)
```

### heatmap only protein bin 29 (control vs. stress)
```{r heatmap only protein bin 29 (control vs. stress)}
# mat_data <- as.matrix(logfc_protein_part)
# distance = dist(mat_data, method = "euclidian")
# cluster = hclust(distance, method = "average")
# my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 299)
# col_breaks = c(seq(-3,-1,length=100), # for blue
#                seq(-1,1,length=100), # for white
#                seq(1,3,length=100)) # for red
# 
# pdf("figures/heatmap_log2FC_protein.pdf", width=6, height=17)
# lmat=rbind(4:3, 2:1)
# lhei=c(0.4, 5)
# lwid=c(1, 4)
# layout(mat = lmat, widths = lwid, heights = lhei)
# heatmap.2(mat_data,
#   #cellnote = signif2,  # same data set for cell labels
#   #notecex=1.5, # numeric scaling factor for cellnote items
#   colsep=1:ncol(mat_data), 
#   rowsep=1:nrow(mat_data),
#   sepwidth=c(0.01,0.05),
#   sepcolor="black",
#   main = "", # heat map title
#   #notecol="black",      # change font color of cell labels to black
#   density.info="none",  # turns off density plot inside color legend
#   trace="none",         # turns off trace lines inside the heat map
#   margins =c(8,18),     # widens margins around plot
#   keysize=1.5,
#   cexRow=1.2,
#   cexCol=1.5,
#   adjCol=c(0,1),
#   srtCol = 330,
#   lhei=c(0.4, 5),
#   col=my_palette,       # use on color palette defined earlier 
#   breaks=col_breaks,    # enable color transition at specified limits
#   #RowSideColors=mycolor,
#   #ColSideColors=column_color,
#   Rowv=as.dendrogram(cluster),
#   dendrogram="row",     # only draw a row dendrogram
#   Colv="NA")            # turn off column clustering```
# 
# dev.off()
```


### Merge logFC and pvalue of different tables (greenhouse)
```{r merge logFC and pvalue of sig. regulated genes greenhouse}
# Stress < Control
res_greenhouse_merge_down_sig <- func_edgeR_top_down(res_greenhouse, res_greenhouse_sensitive, res_greenhouse_tolerant)
dim(res_greenhouse_merge_down_sig) # 749

# Stress > Control
res_greenhouse_merge_up_sig <- func_edgeR_top_up(res_greenhouse, res_greenhouse_sensitive, res_greenhouse_tolerant)
dim(res_greenhouse_merge_up_sig) # 639
head(res_greenhouse_merge_up_sig)

# DOWN
col_plot<-rep(rgb(50,50,50,50,maxColorValue=255),749)
col_plot[res_greenhouse_merge_down_sig$logFC_sens<res_greenhouse_merge_down_sig$logFC_tol]<-rgb(100,0,50,50,maxColorValue=255)
table(col_plot)
plot(res_greenhouse_merge_down_sig$logFC_sens, res_greenhouse_merge_down_sig$logFC_tol, asp=1, col=col_plot)
abline(0,1)
 
# UP
col_plot<-rep(rgb(50,50,50,50,maxColorValue=255),639)
col_plot[res_greenhouse_merge_up_sig$logFC_sens>res_greenhouse_merge_up_sig$logFC_tol]<-rgb(100,0,50,50,maxColorValue=255)
table(col_plot)
#pdf("../figures/logFC_greenhouse_sig_up.pdf", 5, 5)
#pdf("../../../../Doktorarbeit/figures/logFC_greenhouse_sig_up.pdf", width=5, height=5)
par(mar=c(4.5, 4.5, 0.5, 0.5))
plot(res_greenhouse_merge_up_sig$logFC_sens, res_greenhouse_merge_up_sig$logFC_tol, 
     cex.lab = 1.4, cex.axis = 1.2, cex = 2, pch = 19, xlim=c(0,5.5), ylim=c(0,5.5),
     asp=1, col=col_plot, xlab="log2FC of sensitive cultivars", ylab="log2FC of tolerant cultivars")
abline(0,1, col="grey")
legend("bottomright", legend = c(expression(paste("log2FC"[tol], " > log2FC"[sens], ": 160")),
                                 expression(paste("log2FC"[sens], " > log2FC"[tol], ": 479"))),
       cex = 1.1,
       fill=c(rgb(50,50,50,50,maxColorValue=255),
              rgb(100,0,50,50,maxColorValue=255)))
#dev.off()

write.table(res_greenhouse_merge_down_sig, "output/res_greenhouse_merge_down_sig.txt", row.names=F, quote=F, sep="\t")
write.table(res_greenhouse_merge_up_sig, "output/res_greenhouse_merge_up_sig.txt", row.names=F, quote=F, sep="\t")
```


### Merge logFC and pvalue of different tables (field)
```{r merge logFC and pvalue of sig. regulated genes field}
res_field_merge_down_sig <- func_edgeR_top_down(res_field, res_field_sensitive, res_field_tolerant)
dim(res_field_merge_down_sig) #194

res_field_merge_up_sig <- func_edgeR_top_up(res_field, res_field_sensitive, res_field_tolerant)
dim(res_field_merge_up_sig) #267

# DOWN
col_plot<-rep(rgb(50,50,50,50,maxColorValue=255),194)
col_plot[res_field_merge_down_sig$logFC_sens<res_field_merge_down_sig$logFC_tol]<-rgb(100,0,50,50,maxColorValue=255)
table(col_plot)
plot(res_field_merge_down_sig$logFC_sens, res_field_merge_down_sig$logFC_tol, asp=1, col=col_plot)
abline(0,1)

# UP
col_plot<-rep(rgb(50,50,50,50,maxColorValue=255),267)
col_plot[res_field_merge_up_sig$logFC_sens>res_field_merge_up_sig$logFC_tol]<-rgb(100,0,50,50,maxColorValue=255)
table(col_plot)

#pdf("../figures/logFC_field_sig_up.pdf", 5,5)
#pdf("../../../../Doktorarbeit/figures/logFC_field_sig_up.pdf", width=5, height=5)
par(mar=c(4.5, 4.5, 0.5, 0.5))
plot(res_field_merge_up_sig$logFC_sens, res_field_merge_up_sig$logFC_tol, cex.lab=1.4, cex.axis=1.2, cex=2,pch=19,
     asp=1, col=col_plot, xlab="log2FC of sensitive cultivars", ylab="log2FC of tolerant cultivars",
      xlim=c(-2,8), ylim=c(-2,8))
abline(0,1, col="grey")
legend("bottomright", legend = c(expression(paste("log2FC"[tol], " > log2FC"[sens], ": 93")),
                                 expression(paste("log2FC"[sens], " > log2FC"[tol], ": 174"))),
       cex = 1.1,
       fill=c(rgb(50,50,50,50,maxColorValue=255),
              rgb(100,0,50,50,maxColorValue=255)))
#dev.off()

write.table(res_field_merge_down_sig, "output/res_field_merge_down_sig.txt", row.names=F, quote=F, sep="\t")
write.table(res_field_merge_up_sig, "output/res_field_merge_up_sig.txt", row.names=F, quote=F, sep="\t")
```


### Merge logFC and pvalue of greenhouse and field tables
```{r merge logFC and pvalue of sig. regulated genes greenhouse and field}
# DOWN regulated genes
res_greenhouse_field_merge_down_sig <- merge(res_greenhouse_merge_down_sig, # GREENHOUSE
                                             res_field_merge_down_sig,      # FIELD
                                             by.x = "Row.names", by.y = "Row.names", all = T)

rownames(res_greenhouse_field_merge_down_sig) <- res_greenhouse_field_merge_down_sig[,1]
res_greenhouse_field_merge_down_sig <- res_greenhouse_field_merge_down_sig[,-1]

colnames(res_greenhouse_field_merge_down_sig) <- c("logFC_all_greenhouse", "FDR_all_greenhouse",
                                                   "logFC_sens_greenhouse", "FDR_sens_greenhouse", 
                                                   "logFC_tol_greenhouse", "FDR_tol_greenhouse", 
                                                   "func_greenhouse", 
                                                   "logFC_all_field", "FDR_all_field", 
                                                   "logFC_sens_field", "FDR_sens_field", 
                                                   "logFC_tol_field", "FDR_tol_field", 
                                                   "function_field")

# UP regulated genes
res_greenhouse_field_merge_up_sig <- merge(res_greenhouse_merge_up_sig, # GREENHOUSE
                                           res_field_merge_up_sig,      # FIELD
                                           by.x = "Row.names", by.y = "Row.names", all = T)

rownames(res_greenhouse_field_merge_up_sig) <- res_greenhouse_field_merge_up_sig[,1]
res_greenhouse_field_merge_up_sig <- res_greenhouse_field_merge_up_sig[,-1]

colnames(res_greenhouse_field_merge_up_sig) <- c("logFC_all_greenhouse", "FDR_all_greenhouse",
                                                   "logFC_sens_greenhouse", "FDR_sens_greenhouse", 
                                                   "logFC_tol_greenhouse", "FDR_tol_greenhouse", 
                                                   "func_greenhouse", 
                                                   "logFC_all_field", "FDR_all_field", 
                                                   "logFC_sens_field", "FDR_sens_field", 
                                                   "logFC_tol_field", "FDR_tol_field", 
                                                   "function_field")

write.table(res_greenhouse_field_merge_down_sig, "output/res_greenhouse_field_merge_down_sig.txt", row.names=T, quote=F, sep="\t")
write.table(res_greenhouse_field_merge_up_sig, "output/res_greenhouse_field_merge_up_sig.txt", row.names=T, quote=F, sep="\t")
```


### save input tables for CorTo analysis
necessary input for CorTo: PGSC-DMP identifier
therefore, take pageman tables which contain DMP identifier, logFC and dir

#### save input tables for CorTo analysis: sensitive vs. tolerant - greeenhouse
```{r save input tables for CorTo analysis sensitive vs. tolerant - greenhouse}
# control, T > S
res_greenhouse_control_corto_T <- res_greenhouse_control_pageman$pgsc_dmp[which(res_greenhouse_control_pageman$dir == 1)]
# control, S > T
res_greenhouse_control_corto_S <- res_greenhouse_control_pageman$pgsc_dmp[which(res_greenhouse_control_pageman$dir == -1)]
# stress, T > S
res_greenhouse_stress_corto_T <- res_greenhouse_stress_pageman$pgsc_dmp[which(res_greenhouse_stress_pageman$dir == 1)]
# stress, S > T
res_greenhouse_stress_corto_S <- res_greenhouse_stress_pageman$pgsc_dmp[which(res_greenhouse_stress_pageman$dir == -1)]

# intersect between control and stress, T > S
length(intersect(res_greenhouse_stress_corto_T, res_greenhouse_control_corto_T))

# intersect between control and stress, S > T
length(intersect(res_greenhouse_stress_corto_S, res_greenhouse_control_corto_S))

write.table(res_greenhouse_control_corto_T, "corto/corto_input_greenhouse_control_T.txt", row.names=F, quote=F, sep="\t", col.names = F)
write.table(res_greenhouse_control_corto_S, "corto/corto_input_greenhouse_control_S.txt", row.names=F, quote=F, sep="\t", col.names = F)
write.table(res_greenhouse_stress_corto_T, "corto/corto_input_greenhouse_stress_T.txt", row.names=F, quote=F, sep="\t", col.names = F)
write.table(res_greenhouse_stress_corto_S, "corto/corto_input_greenhouse_stress_S.txt", row.names=F, quote=F, sep="\t", col.names = F)
```


#### save input tables for CorTo analysis: sensitive vs. tolerant - field
```{r save input tables for CorTo analysis sensitive vs. tolerant - field}
# control, T > S
res_field_control_corto_T <- res_field_control_pageman$pgsc_dmp[which(res_field_control_pageman$dir == 1)]
# control, S > T
res_field_control_corto_S <- res_field_control_pageman$pgsc_dmp[which(res_field_control_pageman$dir == -1)]
# stress, T > S
res_field_stress_corto_T <- res_field_stress_pageman$pgsc_dmp[which(res_field_stress_pageman$dir == 1)]
# stress, S > T
res_field_stress_corto_S <- res_field_stress_pageman$pgsc_dmp[which(res_field_stress_pageman$dir == -1)]

# intersect between control and stress, T > S
length(intersect(res_field_stress_corto_T, res_field_control_corto_T))

# intersect between control and stress, S > T
length(intersect(res_field_stress_corto_S, res_field_control_corto_S))

write.table(res_field_control_corto_T, "corto/corto_input_field_control_T.txt", row.names=F, quote=F, sep="\t", col.names = F)
write.table(res_field_control_corto_S, "corto/corto_input_field_control_S.txt", row.names=F, quote=F, sep="\t", col.names = F)
write.table(res_field_stress_corto_T, "corto/corto_input_field_stress_T.txt", row.names=F, quote=F, sep="\t", col.names = F)
write.table(res_field_stress_corto_S, "corto/corto_input_field_stress_S.txt", row.names=F, quote=F, sep="\t", col.names = F)
```


#### save input tables for CorTo analysis: control vs. stress - all cultivars together
```{r save input tables for CorTo analysis control vs. stress - all cultivars}
# greenhouse, stress > control (UP)
res_greenhouse_corto_up <- res_greenhouse_pageman$pgsc_dmp[which(res_greenhouse_pageman$dir == 1)]
# greenhouse, control > stress (DOWN)
res_greenhouse_corto_down <- res_greenhouse_pageman$pgsc_dmp[which(res_greenhouse_pageman$dir == -1)]
# field, stress > control (UP)
res_field_corto_up <- res_field_pageman$pgsc_dmp[which(res_field_pageman$dir == 1)]
# field, control > stress (DOWN)
res_field_corto_down <- res_field_pageman$pgsc_dmp[which(res_field_pageman$dir == -1)]


# intersect between greenhouse and field, UP
length(intersect(res_greenhouse_corto_up, res_field_corto_up))

# intersect between greenhouse and field, DOWN
length(intersect(res_greenhouse_corto_down, res_field_corto_down))

write.table(res_greenhouse_corto_up, "corto/corto_input_greenhouse_up.txt", row.names=F, quote=F, sep="\t", col.names = F)
write.table(res_greenhouse_corto_down, "corto/corto_input_greenhouse_down.txt", row.names=F, quote=F, sep="\t", col.names = F)
write.table(res_field_corto_up, "corto/corto_input_field_up.txt", row.names=F, quote=F, sep="\t", col.names = F)
write.table(res_field_corto_down, "corto/corto_input_field_down.txt", row.names=F, quote=F, sep="\t", col.names = F)
# intersection
write.table(intersect(res_greenhouse_corto_up, res_field_corto_up), 
            "corto/corto_input_intersect_greenhouse_field_up.txt", 
            row.names=F, quote=F, sep="\t", col.names = F)
write.table(intersect(res_greenhouse_corto_down, res_field_corto_down), 
            "corto/corto_input_intersect_greenhouse_field_down.txt", 
            row.names=F, quote=F, sep="\t", col.names = F)
```


#### save input tables for CorTo analysis: control vs. stress - greenhouse
```{r save input tables for CorTo analysis control vs. stress - greenhouse}
# tolerant, stress > control (UP)
res_greenhouse_tolerant_corto_up <- res_greenhouse_tolerant_pageman$pgsc_dmp[which(res_greenhouse_tolerant_pageman$dir == 1)]
# tolerant, control > stress (DOWN)
res_greenhouse_tolerant_corto_down <- res_greenhouse_tolerant_pageman$pgsc_dmp[which(res_greenhouse_tolerant_pageman$dir == -1)]
# sensitive, stress > control (UP)
res_greenhouse_sensitive_corto_up <- res_greenhouse_sensitive_pageman$pgsc_dmp[which(res_greenhouse_sensitive_pageman$dir == 1)]
# sensitive, control > stress (DOWN)
res_greenhouse_sensitive_corto_down <- res_greenhouse_sensitive_pageman$pgsc_dmp[which(res_greenhouse_sensitive_pageman$dir == -1)]

# intersect between sensitive and tolerant, UP
length(intersect(res_greenhouse_tolerant_corto_up, res_greenhouse_sensitive_corto_up))

# intersect between sensitive and tolerant, DOWN
length(intersect(res_greenhouse_tolerant_corto_down, res_greenhouse_sensitive_corto_down))

write.table(res_greenhouse_tolerant_corto_up, "corto/corto_input_greenhouse_tolerant_up.txt", row.names=F, quote=F, sep="\t", col.names = F)
write.table(res_greenhouse_tolerant_corto_down, "corto/corto_input_greenhouse_tolerant_down.txt", row.names=F, quote=F, sep="\t", col.names = F)
write.table(res_greenhouse_sensitive_corto_up, "corto/corto_input_greenhouse_sensitive_up.txt", row.names=F, quote=F, sep="\t", col.names = F)
write.table(res_greenhouse_sensitive_corto_down, "corto/corto_input_greenhouse_sensitive_down.txt", row.names=F, quote=F, sep="\t", col.names = F)
```


#### save input tables for CorTo analysis: control vs. stress - field
```{r save input tables for CorTo analysis control vs. stress - field}
# tolerant, stress > control (UP)
res_field_tolerant_corto_up <- res_field_tolerant_pageman$pgsc_dmp[which(res_field_tolerant_pageman$dir == 1)]
# tolerant, control > stress (DOWN)
res_field_tolerant_corto_down <- res_field_tolerant_pageman$pgsc_dmp[which(res_field_tolerant_pageman$dir == -1)]
# sensitive, stress > control (UP)
res_field_sensitive_corto_up <- res_field_sensitive_pageman$pgsc_dmp[which(res_field_sensitive_pageman$dir == 1)]
# sensitive, control > stress (DOWN)
res_field_sensitive_corto_down <- res_field_sensitive_pageman$pgsc_dmp[which(res_field_sensitive_pageman$dir == -1)]

# intersect between sensitive and tolerant, UP
length(intersect(res_field_tolerant_corto_up, res_field_sensitive_corto_up))

# intersect between sensitive and tolerant, DOWN
length(intersect(res_field_tolerant_corto_down, res_field_sensitive_corto_down))

write.table(res_field_tolerant_corto_up, "corto/corto_input_field_tolerant_up.txt", row.names=F, quote=F, sep="\t", col.names = F)
write.table(res_field_tolerant_corto_down, "corto/corto_input_field_tolerant_down.txt", row.names=F, quote=F, sep="\t", col.names = F)
write.table(res_field_sensitive_corto_up, "corto/corto_input_field_sensitive_up.txt", row.names=F, quote=F, sep="\t", col.names = F)
write.table(res_field_sensitive_corto_down, "corto/corto_input_field_sensitive_down.txt", row.names=F, quote=F, sep="\t", col.names = F)
```


### Save final DGE list with IDs (DMP, DMG), logFC, FDR, PGSC function, dir, MapMan BinCode, MapMan BinName and MapMan Description
#### Save final DGE list: tolerant vs. sensitive
```{r save final DGE list tolerant vs. sensitive}

# dir ==  1 --> tol > sens --> T
final_greenhouse_control_T <- func_final_DGE_list(res = res_greenhouse_control, dir_value = 1)
final_greenhouse_stress_T <- func_final_DGE_list(res = res_greenhouse_stress, dir_value = 1)
final_field_control_T <- func_final_DGE_list(res = res_field_control, dir_value = 1)
final_field_stress_T <- func_final_DGE_list(res = res_field_stress, dir_value = 1)

write.table(final_greenhouse_control_T, "output/final_dge_greenhouse_control_T.txt", sep="\t", row.names=F, quote=F)
write.table(final_greenhouse_stress_T, "output/final_dge_greenhouse_stress_T.txt", sep="\t", row.names=F, quote=F)
write.table(final_field_control_T, "output/final_dge_field_control_T.txt", sep="\t", row.names=F, quote=F)
write.table(final_field_stress_T, "output/final_dge_field_stress_T.txt", sep="\t", row.names=F, quote=F)

# dir == -1 --> sens > tol --> S
final_greenhouse_control_S <- func_final_DGE_list(res = res_greenhouse_control, dir_value = -1)
final_greenhouse_stress_S <- func_final_DGE_list(res = res_greenhouse_stress, dir_value = -1)
final_field_control_S <- func_final_DGE_list(res = res_field_control, dir_value = -1)
final_field_stress_S <- func_final_DGE_list(res = res_field_stress, dir_value = -1)

write.table(final_greenhouse_control_S, "output/final_dge_greenhouse_control_S.txt", sep="\t", row.names=F, quote=F)
write.table(final_greenhouse_stress_S, "output/final_dge_greenhouse_stress_S.txt", sep="\t", row.names=F, quote=F)
write.table(final_field_control_S, "output/final_dge_field_control_S.txt", sep="\t", row.names=F, quote=F)
write.table(final_field_stress_S, "output/final_dge_field_stress_S.txt", sep="\t", row.names=F, quote=F)
```


#### Save final DGE list: control vs. stress
```{r save final DGE list control vs. stress}

# dir ==  1 --> stress > control --> up
final_greenhouse_all_up <- func_final_DGE_list(res = res_greenhouse, dir_value = 1) # all cultivars together
final_greenhouse_tolerant_up <- func_final_DGE_list(res = res_greenhouse_tolerant, dir_value = 1)
final_greenhouse_sensitive_up <- func_final_DGE_list(res = res_greenhouse_sensitive, dir_value = 1)
final_field_all_up <- func_final_DGE_list(res = res_field, dir_value = 1) # all cultivars together
final_field_tolerant_up <- func_final_DGE_list(res = res_field_tolerant, dir_value = 1)
final_field_sensitive_up <- func_final_DGE_list(res = res_field_sensitive, dir_value = 1)

write.table(final_greenhouse_all_up, "output/final_dge_greenhouse_all_up.txt", sep="\t", row.names=F, quote=F)
write.table(final_greenhouse_tolerant_up, "output/final_dge_greenhouse_tolerant_up.txt", sep="\t", row.names=F, quote=F)
write.table(final_greenhouse_sensitive_up, "output/final_dge_greenhouse_sensitive_up.txt", sep="\t", row.names=F, quote=F)
write.table(final_field_all_up, "output/final_dge_field_all_up.txt", sep="\t", row.names=F, quote=F)
write.table(final_field_tolerant_up, "output/final_dge_field_tolerant_up.txt", sep="\t", row.names=F, quote=F)
write.table(final_field_sensitive_up, "output/final_dge_field_sensitive_up.txt", sep="\t", row.names=F, quote=F)

# dir == -1 --> control > stress --> down
final_greenhouse_all_down <- func_final_DGE_list(res = res_greenhouse, dir_value = -1) # all cultivars together
final_greenhouse_tolerant_down <- func_final_DGE_list(res = res_greenhouse_tolerant, dir_value = -1)
final_greenhouse_sensitive_down <- func_final_DGE_list(res = res_greenhouse_sensitive, dir_value = -1)
final_field_all_down <- func_final_DGE_list(res = res_field, dir_value = -1) # all cultivars together
final_field_tolerant_down <- func_final_DGE_list(res = res_field_tolerant, dir_value = -1)
final_field_sensitive_down <- func_final_DGE_list(res = res_field_sensitive, dir_value = -1)

write.table(final_greenhouse_all_down, "output/final_dge_greenhouse_all_down.txt", sep="\t", row.names=F, quote=F)
write.table(final_greenhouse_tolerant_down, "output/final_dge_greenhouse_tolerant_down.txt", sep="\t", row.names=F, quote=F)
write.table(final_greenhouse_sensitive_down, "output/final_dge_greenhouse_sensitive_down.txt", sep="\t", row.names=F, quote=F)
write.table(final_field_all_down, "output/final_dge_field_all_down.txt", sep="\t", row.names=F, quote=F)
write.table(final_field_tolerant_down, "output/final_dge_field_tolerant_down.txt", sep="\t", row.names=F, quote=F)
write.table(final_field_sensitive_down, "output/final_dge_field_sensitive_down.txt", sep="\t", row.names=F, quote=F)
```


#### Merge final DGE list: control vs. stress (compare greenhouse and field)
**Merge results of the same comparison for greenhouse and field**

```{r merge final DGE list control vs. stress (compare greenhouse and field)}

final_all_up <- func_combine_two_final_DGE_lists_2(final_DGE_list_1 = final_greenhouse_all_up,
                                                   final_DGE_list_2 = final_field_all_up)

final_all_down <- func_combine_two_final_DGE_lists_2(final_DGE_list_1 = final_greenhouse_all_down,
                                                     final_DGE_list_2 = final_field_all_down)

write.table(final_all_up, "output/final_dge_all_up_merged_by_dmg.txt", sep="\t", row.names=F, quote=F)
write.table(final_all_down, "output/final_dge_all_down_merged_by_dmg.txt", sep="\t", row.names=F, quote=F)
```

#### Merge final DGE list: control vs. stress (compare tolerant and sensitive)
**Merge results of the same comparison for tolerant and sensitive, e.g. greenhouse up-regulated genes**


```{r merge final DGE list control vs. stress (compare tolerant and sensitive)}

final_greenhouse_up_merged_by_dmg <- func_combine_two_final_DGE_lists(final_greenhouse_tolerant_up,
                                                                      final_greenhouse_sensitive_up)

final_greenhouse_down_merged_by_dmg <- func_combine_two_final_DGE_lists(final_greenhouse_tolerant_down,
                                                                        final_greenhouse_sensitive_down)

final_field_up_merged_by_dmg <- func_combine_two_final_DGE_lists(final_field_tolerant_up,
                                                                 final_field_sensitive_up)

final_field_down_merged_by_dmg <- func_combine_two_final_DGE_lists(final_field_tolerant_down,
                                                                   final_field_sensitive_down)

write.table(final_greenhouse_up_merged_by_dmg, "output/final_dge_greenhouse_up_merged_by_dmg.txt", sep="\t", row.names=F, quote=F)
write.table(final_greenhouse_down_merged_by_dmg, "output/final_dge_greenhouse_down_merged_by_dmg.txt", sep="\t", row.names=F, quote=F)
write.table(final_field_up_merged_by_dmg, "output/final_dge_field_up_merged_by_dmg.txt", sep="\t", row.names=F, quote=F)
write.table(final_field_down_merged_by_dmg, "output/final_dge_field_down_merged_by_dmg.txt", sep="\t", row.names=F, quote=F)
```


```{r try out something}
final_greenhouse_tolerant_up2 <- droplevels(final_greenhouse_tolerant_up)
table(final_greenhouse_tolerant_up2$BINCODE)

try_out <- cbind("tol_up" = table(final_greenhouse_tolerant_up$NAME),
                 "sens_up" = table(final_greenhouse_sensitive_up$NAME),
                 "tol_down" = table(final_greenhouse_tolerant_down$NAME),
                 "sens_down" = table(final_greenhouse_sensitive_down$NAME))
head(try_out)

head(mapman_bincode_merge_ordered)

try_out_2 <- merge(mapman_bincode_merge_ordered, try_out, all.x = T, all.y = F, by.x = "NAME", by.y = "row.names")
head(try_out_2)
dim(try_out_2)
```


# generate data for Joachim
## table of identifier (protein-coding genes and non-coding genes) with functional annotation, but only for 25846 selected genes
```{r generate data for Joachim}
# functional annotation of non-coding genes
assoc_non_coding_genes <- read.table("data/non-protein-coding-genes.txt", header=F, sep="\t")
# use only two columns
assoc_non_coding_genes2 <- assoc_non_coding_genes[, c("V9", "V13")]
colnames(assoc_non_coding_genes2) <- c("identifier", "func")
head(assoc_non_coding_genes2)

# only unique annotation per gene (DMG)
assoc_pgsc_only_dmg <- unique( assoc_pgsc[, c("pgsc_dmg", "func")] )
colnames(assoc_pgsc_only_dmg) <- c("identifier", "func")
head(assoc_pgsc_only_dmg)

# combine table of identifier for protein-coding genes and non-coding genes
assoc_pgsc_all <- rbind(assoc_pgsc_only_dmg, assoc_non_coding_genes2)
head(assoc_pgsc_all)

# Identifier of selected genes
head(rownames(counts_keep))
counts_keep_rownames <- rownames(counts_keep)

# select only part of the annotation table with selected genes
length(which(assoc_pgsc_all$identifier %in% counts_keep_rownames))
assoc_pgsc_keep <- assoc_pgsc_all[which(assoc_pgsc_all$identifier %in% counts_keep_rownames), ]

write.table(assoc_pgsc_keep, "output/assoc_pgsc_keep.txt", sep="\t", row.names = F)
```



# save workspace
```{r save workspace}
save.image("exp_counts_edgeR.RData")
```

